
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Manual &#8212; MView</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinxdoc.css" />
    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script async="async" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Input formats" href="formats_in.html" />
    <link rel="prev" title="Contents" href="../contents.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="formats_in.html" title="Input formats"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="../contents.html" title="Contents"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">MView</a> &#187;</li>
          <li class="nav-item nav-item-1"><a href="../contents.html" accesskey="U">Contents</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Manual</a></li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <LINK rel="stylesheet" type="text/css" href="../_static/MView.css"><section id="manual">
<h1>Manual<a class="headerlink" href="#manual" title="Permalink to this headline">¶</a></h1>
<section id="basic-usage">
<h2>Basic usage<a class="headerlink" href="#basic-usage" title="Permalink to this headline">¶</a></h2>
<section id="getting-started">
<h3>Getting started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h3>
<p>Given an existing sequence alignment in a file <code class="docutils literal notranslate"><span class="pre">data.dat</span></code> in FASTA format,
the simplest usage might be:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="ow">in</span> <span class="n">fasta</span> <span class="n">data</span><span class="o">.</span><span class="n">dat</span>
</pre></div>
</div>
<p>Similarly, if the input file contained a CLUSTAL alignment:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="ow">in</span> <span class="n">clustal</span> <span class="n">data</span><span class="o">.</span><span class="n">dat</span>
</pre></div>
</div>
<p>In either case, the output would be a stacked alignment with extra columns
added to show row numbers and percent coverage and percent identity (with
respect to the first sequence, see <a class="reference internal" href="#ref-cov-pid"><span class="std std-ref">Coverage and identity</span></a>), looking something
like this, regardless of the input format:</p>
<PRE>
Reference sequence (1): EGFR_HUMAN
Identities normalised by aligned length.

                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIPVAIKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC
2 PR2_DROME    97.1%  35.7%    ISVNKQLGTGEFGIVQQGVWSNGNE-----------RIQVAIKCLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV
3 ITK_HUMAN    90.0%  32.9%    LTFVQEIGSGQFGLVHLGYWLN--------------KDKVAIKTIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC
4 PTK7_HUMAN   97.1%  21.2%    IREVKQIGVGQFGAVVLAEMTGLS-XLPKGSMNADGVALVAVKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC
5 KIN31_CAEEL 100.0%  31.5%    VELTKKLGEGAFGEVWKGKLLKILDA-------NHQPVLVAVKTAKLESMTKEQIKEIMREARLMRNLDHINVVKFFGVA
</PRE><p>To process the output of a BLAST run use something like:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="ow">in</span> <span class="n">blast</span> <span class="n">blastresults</span><span class="o">.</span><span class="n">dat</span>
</pre></div>
</div>
<p>while to process the output of a FASTA run (the sequence database search
program from U. of Virginia, not the simple FASTA/Pearson data format) use
something like:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="ow">in</span> <span class="n">uvfasta</span> <span class="n">fastaresults</span><span class="o">.</span><span class="n">dat</span>
</pre></div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">-in</span></code> option isn’t always necessary. If the filename extension, or the
filename itself minus any directory path begins with or contains the first few
letters of the valid <code class="docutils literal notranslate"><span class="pre">-in</span></code> options (e.g., <code class="docutils literal notranslate"><span class="pre">mydata.msf</span></code> or <code class="docutils literal notranslate"><span class="pre">mydata.fasta</span></code>
or <code class="docutils literal notranslate"><span class="pre">tfastx_run.dat</span></code>), MView tries to choose a sensible input format,
allowing multiple files in mixed formats to be supplied on the command
line. The <code class="docutils literal notranslate"><span class="pre">-in</span></code> option will always override this mechanism but requires that
all input files be of the same format.</p>
</section>
<section id="the-ruler-line">
<span id="ref-rulers"></span><h3>The ruler line<a class="headerlink" href="#the-ruler-line" title="Permalink to this headline">¶</a></h3>
<p>The ruler line at the top is enabled by default. To remove it, use the
<code class="docutils literal notranslate"><span class="pre">-ruler</span> <span class="pre">off</span></code> option:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="ow">in</span> <span class="n">fasta</span> <span class="o">-</span><span class="n">ruler</span> <span class="n">off</span> <span class="n">data</span><span class="o">.</span><span class="n">dat</span>
</pre></div>
</div>
<p>gives:</p>
<PRE>
Reference sequence (1): EGFR_HUMAN
Identities normalised by aligned length.

1 EGFR_HUMAN  100.0% 100.0%    FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIPVAIKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC
2 PR2_DROME    97.1%  35.7%    ISVNKQLGTGEFGIVQQGVWSNGNE-----------RIQVAIKCLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV
3 ITK_HUMAN    90.0%  32.9%    LTFVQEIGSGQFGLVHLGYWLN--------------KDKVAIKTIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC
4 PTK7_HUMAN   97.1%  21.2%    IREVKQIGVGQFGAVVLAEMTGLS-XLPKGSMNADGVALVAVKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC
5 KIN31_CAEEL 100.0%  31.5%    VELTKKLGEGAFGEVWKGKLLKILDA-------NHQPVLVAVKTAKLESMTKEQIKEIMREARLMRNLDHINVVKFFGVA
</PRE><p>Removing the ruler also removes some column header information, which could
cause confusion.</p>
<p>Only one kind of ruler is currently provided, numbering the columns of the
final alignment from M to N (incrementing) or N to M (decrementing) based on
the input sequence numbering, if any. For multiple alignments like the one
above with no numbering the ruler runs from 1 to the length of the alignment.</p>
<p>For database searches that translate nucleotide sequences to protein, such as
TBLASTX, the rulers differ slightly in that the native query numbering is
given in nucleotide units, but MView reports amino acid units instead (using
modulo 3 arithmetic).</p>
</section>
<section id="command-line-options">
<h3>Command line options<a class="headerlink" href="#command-line-options" title="Permalink to this headline">¶</a></h3>
<p>ALl available options can be listed using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="n">help</span>
</pre></div>
</div>
<p>There are a lot of options, but the main ones are described in this manual.
Many of them can be abbreviated as long as they are unambiguous, for example,
the option <code class="docutils literal notranslate"><span class="pre">-reference</span></code> can be shortened to <code class="docutils literal notranslate"><span class="pre">-ref</span></code>.</p>
</section>
</section>
<section id="layout-and-filtering">
<h2>Layout and filtering<a class="headerlink" href="#layout-and-filtering" title="Permalink to this headline">¶</a></h2>
<section id="pagination">
<h3>Pagination<a class="headerlink" href="#pagination" title="Permalink to this headline">¶</a></h3>
<p>The default layout is a single unbroken horizontal band of alignment - fine if
scrolling inside Firefox. However, you may prefer to break the alignment into
vertically stacked panes. For panes, for example, 80 columns wide, set
<code class="docutils literal notranslate"><span class="pre">-width</span> <span class="pre">80</span></code>. Widths refer to the alignment, not to the whole displayed
output.</p>
</section>
<section id="column-ranges">
<h3>Column ranges<a class="headerlink" href="#column-ranges" title="Permalink to this headline">¶</a></h3>
<p>It is possible to narrow (or expand) the displayed range of columns of the
alignment, for example, <code class="docutils literal notranslate"><span class="pre">-range</span> <span class="pre">10:78</span></code> would select only that column range
using the numbering scheme reported when <code class="docutils literal notranslate"><span class="pre">-ruler</span> <span class="pre">on</span></code> is set (see
<a class="reference internal" href="#ref-rulers"><span class="std std-ref">The ruler line</span></a>). Note: the range setting is not related to the sequence
position labelling for blast/fasta database search input; it’s just the
position along the ruler.</p>
<p>The order of the numbers is unimportant making it simpler to state interest in
a region of the alignment that might actually be reversed in the output (e.g.,
a BLASTN search hit matching the reverse complement of the query strand).</p>
</section>
<section id="changing-the-reference-sequence">
<span id="ref-reference-row"></span><h3>Changing the reference sequence<a class="headerlink" href="#changing-the-reference-sequence" title="Permalink to this headline">¶</a></h3>
<p>One can colour and compute identities with respect to a sequence other than
the first/query sequence using the <code class="docutils literal notranslate"><span class="pre">-reference</span></code> option.</p>
<p>This takes either the sequence identifier or an integer argument corresponding
to the ranking or ordering of a sequence usually shown in the first labelling
column of MView output. For multiple alignment input formats, sequences are
numbered from 1, while for searches the hits are numbered from 1, but the
query itself is 0, so beware.</p>
</section>
<section id="sorting-by-percent-coverage-or-percent-identity">
<span id="ref-sorting"></span><h3>Sorting by percent coverage or percent identity<a class="headerlink" href="#sorting-by-percent-coverage-or-percent-identity" title="Permalink to this headline">¶</a></h3>
<p>By default MView outputs sequences in the same order they were read in and
computes percent coverage and percent identity values (see
<a class="reference internal" href="#ref-cov-pid"><span class="std std-ref">Coverage and identity</span></a>).</p>
<p>You can change the ordering with the <code class="docutils literal notranslate"><span class="pre">-sort</span></code> option:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">-</span><span class="n">sort</span> <span class="n">cov</span>
<span class="o">-</span><span class="n">sort</span> <span class="n">pid</span>
<span class="o">-</span><span class="n">sort</span> <span class="n">cov</span><span class="p">:</span><span class="n">pid</span>
<span class="o">-</span><span class="n">sort</span> <span class="n">pid</span><span class="p">:</span><span class="n">cov</span>
</pre></div>
</div>
<p>to sort the output (descending) by coverage, percent identity, coverage then
percent identity, or percent identity then coverage. Rows that coincide in the
sort retain their original local ordering (ascending row number).</p>
<p>You can also change the reference sequence (see <a class="reference internal" href="#ref-reference-row"><span class="std std-ref">Changing the reference sequence</span></a>) and
apply one of these sorts like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">-</span><span class="n">ref</span> <span class="mi">2</span> <span class="o">-</span><span class="n">sort</span> <span class="n">cov</span><span class="p">:</span><span class="n">pid</span>
</pre></div>
</div>
<p>which would compute all the coverage and percent identities with respect to
row 2, then sort by coverage and percent identity placing row 2 first.</p>
<p>Default ordering taken from the input alignment:</p>
<PRE>
Reference sequence (1): EGFR_HUMAN
Identities normalised by aligned length.

1 EGFR_HUMAN  100.0% 100.0%  FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIPVAIKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC
2 PR2_DROME    97.1%  35.7%  ISVNKQLGTGEFGIVQQGVWSNGNE-----------RIQVAIKCLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV
3 ITK_HUMAN    90.0%  32.9%  LTFVQEIGSGQFGLVHLGYWLN--------------KDKVAIKTIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC
4 PTK7_HUMAN   97.1%  21.2%  IREVKQIGVGQFGAVVLAEMTGLS-XLPKGSMNADGVALVAVKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC
5 KIN31_CAEEL 100.0%  31.5%  VELTKKLGEGAFGEVWKGKLLKILDA-------NHQPVLVAVKTAKLESMTKEQIKEIMREARLMRNLDHINVVKFFGVA
</PRE><p>After sorting by coverage and percent identity to row 2:</p>
<PRE>
Reference sequence (2): PR2_DROME
Identities normalised by aligned length.

2 PR2_DROME   100.0% 100.0%  ISVNKQLGTGEFGIVQQGVWSNGNE-----------RIQVAIKCLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV
1 EGFR_HUMAN  100.0%  35.7%  FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIPVAIKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC
5 KIN31_CAEEL 100.0%  30.1%  VELTKKLGEGAFGEVWKGKLLKILDA-------NHQPVLVAVKTAKLESMTKEQIKEIMREARLMRNLDHINVVKFFGVA
4 PTK7_HUMAN   97.1%  25.0%  IREVKQIGVGQFGAVVLAEMTGLS-XLPKGSMNADGVALVAVKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC
3 ITK_HUMAN    92.6%  33.8%  LTFVQEIGSGQFGLVHLGYWLN--------------KDKVAIKTIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC
</PRE></section>
<section id="filtering-rows">
<span id="ref-filtering-rows"></span><h3>Filtering rows<a class="headerlink" href="#filtering-rows" title="Permalink to this headline">¶</a></h3>
<p><strong>Showing only the top N rows</strong></p>
<p>Usually, specifying a limited number of hits to view from a long search
alignment speeds things up a lot as there’s less parsing and less formatting
to be generated, so to get the best 10 hits, use the option <code class="docutils literal notranslate"><span class="pre">-top</span> <span class="pre">10</span></code>.</p>
<p><strong>Filtering by percent identity</strong></p>
<p>The <code class="docutils literal notranslate"><span class="pre">-minident</span> <span class="pre">N</span></code> option will report only those hits above some threshold
percent identity N compared to the reference row; useful for looking for close
matches to the query or other reference sequence.</p>
<p>Similarly, you can exclude the strong matches using <code class="docutils literal notranslate"><span class="pre">-maxident</span> <span class="pre">N</span></code>.</p>
<p>Both options can be combined.</p>
<p><strong>Showing and hiding sets of rows</strong></p>
<p>Rows can be dropped explicitly using the <code class="docutils literal notranslate"><span class="pre">-hide</span></code> option. This can be
supplied a comma-separated list of row identifiers, rank numbers, rank number
ranges (1,2,3, 1..3, 1:3 are all equivalent), regular expressions (case
insensitive, enclosed between // characters) to match against row identifiers,
or the <code class="docutils literal notranslate"><span class="pre">*</span></code> symbol meaning all rows.</p>
<p>Likewise, the <code class="docutils literal notranslate"><span class="pre">-show</span></code> option specifies a list of rows to keep in the
alignment. The <code class="docutils literal notranslate"><span class="pre">-show</span></code> option overrides <code class="docutils literal notranslate"><span class="pre">-hide</span></code> whenever a row is common
to both.</p>
<p>For example, the options:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">-</span><span class="n">hide</span> <span class="nb">all</span>  <span class="o">-</span><span class="n">show</span> <span class="s1">&#39;2,3,6..10,/^pdb/&#39;</span>
</pre></div>
</div>
<p>or even:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">-</span><span class="n">hide</span> <span class="s1">&#39;/.*/&#39;</span>  <span class="o">-</span><span class="n">show</span> <span class="s1">&#39;2,3,6:10,/^pdb/&#39;</span>
</pre></div>
</div>
<p>would hide everything except rows 2, 3, 6 through 10 inclusive, and any hits
beginning with the string ‘pdb’.</p>
<p>Note: the currently set reference row is still used for percent identity and
colouring operations, even though the row may have been dropped from display
by the <code class="docutils literal notranslate"><span class="pre">-hide</span></code> list (see <a class="reference internal" href="#ref-reference-row"><span class="std std-ref">Changing the reference sequence</span></a>).</p>
<p><strong>Data format specific filters</strong></p>
<p>Other filters specific to BLASTP, FASTA, etc., input formats allow cutoffs on
scores or p-values, etc. In particular, it is possible to apply some control
over the selection of HSPs used in building the MView alignment using the
<code class="docutils literal notranslate"><span class="pre">-hsp</span></code> filtering option.</p>
<p>Some search programs produce DNA strand-directional output (e.g., BLASTN) and
you can extract or output the results separately. For example, to see just the
plus strand matches:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="ow">in</span> <span class="n">blast</span> <span class="o">-</span><span class="n">strand</span> <span class="n">p</span> <span class="n">blastn_results</span><span class="o">.</span><span class="n">dat</span>
</pre></div>
</div>
<p>The choices are <code class="docutils literal notranslate"><span class="pre">p</span></code>, <code class="docutils literal notranslate"><span class="pre">m</span></code>, <code class="docutils literal notranslate"><span class="pre">both</span></code>.</p>
<p>Of interest to anyone using PSI-BLAST, you can display alignments for any/all
iterations of a PSI-BLAST run using, say:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="ow">in</span> <span class="n">blast</span> <span class="o">-</span><span class="n">cycle</span> <span class="mi">1</span><span class="p">,</span><span class="n">last</span> <span class="n">psiblast_results</span><span class="o">.</span><span class="n">dat</span>
</pre></div>
</div>
<p>to get just those two iterations. The default is to display only the last
iteration. If you want all output, use <code class="docutils literal notranslate"><span class="pre">-cycle</span> <span class="pre">all</span></code>.</p>
<p><strong>Keeping rows, but ignoring them in calculations</strong></p>
<p>Another control option can be used to prevent MView from using rows for
colouring or for calculation of percent identities although these rows will
still be displayed. Use <code class="docutils literal notranslate"><span class="pre">-nop</span></code> to specify a list (comma-separated as usual)
of identifiers or row numbers to flag for “NO Processing”.</p>
<p><strong>Comment lines, another way to ignore rows</strong></p>
<p>If a sequence identifier starts with the <code class="docutils literal notranslate"><span class="pre">#</span></code> character, it is treated as a
<code class="docutils literal notranslate"><span class="pre">nop</span></code> row, i.e., as a comment. This is useful for displaying other textual
information (e.g., secondary structure predictions) within the alignment.</p>
<p>Here is an example with two lines of input data beginning with dummy
identifiers <code class="docutils literal notranslate"><span class="pre">#residues</span></code> and <code class="docutils literal notranslate"><span class="pre">#properties</span></code>:</p>
<PRE>
                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIPVAIKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC
2 PR2_DROME    97.1%  35.7%    ISVNKQLGTGEFGIVQQGVWSNGNE-----------RIQVAIKCLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV
3 ITK_HUMAN    90.0%  32.9%    LTFVQEIGSGQFGLVHLGYWLN--------------KDKVAIKTIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC
4 PTK7_HUMAN   97.1%  21.2%    IREVKQIGVGQFGAVVLAEMTGLS-XLPKGSMNADGVALVAVKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC
5 KIN31_CAEEL 100.0%  31.5%    VELTKKLGEGAFGEVWKGKLLKILDA-------NHQPVLVAVKTAKLESMTKEQIKEIMREARLMRNLDHINVVKFFGVA
  residues                     ------------------------------------------^Lys--------------^Glu----------------
  properties                   ------------------------------------^^^^^^hydrophobic---------------------------
</PRE><p>Note 1: MView elides the hash character in the output and also the row
numbers. However, the name or row number based filtering mechanism described
above still works.</p>
<p>Note 2: Comment lines like this cannot contain whitespace in the sequence
region - they look like sequences and are processed as such by MView.</p>
</section>
<section id="labels-annotations-and-sequences">
<h3>Labels, annotations and sequences<a class="headerlink" href="#labels-annotations-and-sequences" title="Permalink to this headline">¶</a></h3>
<p>You can switch off some of the displayed columns, referred to here as
labels. Labels are in blocks numbered from zero (perverse, but the original
reasoning was that the input data starts with the sequence identifiers in
column 1 and MView tacks on a rank number in front, so make that column 0).</p>
<blockquote>
<div><table class="docutils align-default">
<colgroup>
<col style="width: 11%" />
<col style="width: 89%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Column</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>0</p></td>
<td><p>rank</p></td>
</tr>
<tr class="row-odd"><td><p>1</p></td>
<td><p>identifier</p></td>
</tr>
<tr class="row-even"><td><p>2</p></td>
<td><p>description</p></td>
</tr>
<tr class="row-odd"><td><p>3</p></td>
<td><p>score block (may contain many score columns)</p></td>
</tr>
<tr class="row-even"><td><p>4</p></td>
<td><p>percent coverage</p></td>
</tr>
<tr class="row-odd"><td><p>5</p></td>
<td><p>percent identity</p></td>
</tr>
<tr class="row-even"><td><p>6</p></td>
<td><p>query sequence positions (blast or fasta searches)</p></td>
</tr>
<tr class="row-odd"><td><p>7</p></td>
<td><p>hit sequence positions (blast or fasta searches)</p></td>
</tr>
<tr class="row-even"><td><p>8</p></td>
<td><p>trailing optional fields (blast -outfmt 7 data)</p></td>
</tr>
</tbody>
</table>
</div></blockquote>
<p>Any of the of the label types can be switched off with an option like
<code class="docutils literal notranslate"><span class="pre">-label2</span></code> to remove the descriptions label at column 2, and so on.</p>
<p>You can also disable processing and display of the sequences with the
<code class="docutils literal notranslate"><span class="pre">-sequences</span> <span class="pre">off</span></code> option. This would be useful if you only want to see
summarized scoring information from many blast runs, for example.</p>
</section>
</section>
<section id="adding-html">
<h2>Adding HTML<a class="headerlink" href="#adding-html" title="Permalink to this headline">¶</a></h2>
<section id="basic-html">
<h3>Basic HTML<a class="headerlink" href="#basic-html" title="Permalink to this headline">¶</a></h3>
<p>To add some HTML markup a few extra options are needed, for example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="ow">in</span> <span class="n">fasta</span> <span class="o">-</span><span class="n">html</span> <span class="n">head</span> <span class="n">data</span><span class="o">.</span><span class="n">dat</span> <span class="o">&gt;</span> <span class="n">data</span><span class="o">.</span><span class="n">html</span>
</pre></div>
</div>
<p>produces a complete page of HTML and you can load this into your Web browser
with a URL like <code class="docutils literal notranslate"><span class="pre">file:///full/path/to/the/folder/data.html</span></code>.</p>
<p>To colour all the residues using the default built-in colourmap for proteins:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="ow">in</span> <span class="n">fasta</span> <span class="o">-</span><span class="n">html</span> <span class="n">head</span> <span class="o">-</span><span class="n">coloring</span> <span class="nb">any</span> <span class="n">data</span><span class="o">.</span><span class="n">dat</span> <span class="o">&gt;</span> <span class="n">data</span><span class="o">.</span><span class="n">html</span>
</pre></div>
</div>
<p>produces:</p>
<PRE>
Reference sequence (1): EGFR_HUMAN
Identities normalised by aligned length.
Colored by: property

                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    <SPAN style="color:#009900">F</SPAN><SPAN style="color:#cc0000">KK</SPAN><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">VLG</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#33cc00">GA</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">G</SPAN><SPAN style="color:#0099ff">T</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#009900">Y</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">GL</SPAN><SPAN style="color:#009900">W</SPAN><SPAN style="color:#33cc00">IP</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">G</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#666666">---------</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">IPVAI</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">A</SPAN><SPAN style="color:#0099ff">TS</SPAN><SPAN style="color:#33cc00">P</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#666666">-</SPAN><SPAN style="color:#33cc00">A</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">IL</SPAN><SPAN style="color:#0033ff">DE</SPAN><SPAN style="color:#33cc00">A</SPAN><SPAN style="color:#009900">Y</SPAN><SPAN style="color:#33cc00">VMA</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#33cc00">P</SPAN><SPAN style="color:#009900">H</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#ffff00">C</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#33cc00">LLGI</SPAN><SPAN style="color:#ffff00">C</SPAN>
2 PR2_DROME    97.1%  35.7%    <SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#33cc00">LG</SPAN><SPAN style="color:#0099ff">T</SPAN><SPAN style="color:#33cc00">G</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">GIV</SPAN><SPAN style="color:#6600cc">QQ</SPAN><SPAN style="color:#33cc00">GV</SPAN><SPAN style="color:#009900">W</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#33cc00">G</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#666666">-----------</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#33cc00">VAI</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#ffff00">C</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#ffff00">C</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#33cc00">M</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#666666">-</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#33cc00">PM</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">AAIM</SPAN><SPAN style="color:#009900">H</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#009900">H</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#33cc00">IV</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#009900">Y</SPAN><SPAN style="color:#33cc00">GVV</SPAN>
3 ITK_HUMAN    90.0%  32.9%    <SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#0099ff">T</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">IG</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#33cc00">G</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">GLV</SPAN><SPAN style="color:#009900">H</SPAN><SPAN style="color:#33cc00">LG</SPAN><SPAN style="color:#009900">YW</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#666666">--------------</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">VAI</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0099ff">T</SPAN><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">GAM</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#666666">---</SPAN><SPAN style="color:#0033ff">EED</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#0033ff">EE</SPAN><SPAN style="color:#33cc00">A</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">VMM</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#009900">H</SPAN><SPAN style="color:#33cc00">P</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">LV</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#009900">Y</SPAN><SPAN style="color:#33cc00">GV</SPAN><SPAN style="color:#ffff00">C</SPAN>
4 PTK7_HUMAN   97.1%  21.2%    <SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#33cc00">IGVG</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">GAVVLA</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">M</SPAN><SPAN style="color:#0099ff">T</SPAN><SPAN style="color:#33cc00">GL</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#666666">-X</SPAN><SPAN style="color:#33cc00">LP</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">G</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#33cc00">M</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#33cc00">A</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#33cc00">GVALVAV</SPAN><SPAN style="color:#cc0000">KK</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">P</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#666666">-</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">VL</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">M</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#009900">H</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#33cc00">IV</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#33cc00">LLAI</SPAN><SPAN style="color:#ffff00">C</SPAN>
5 KIN31_CAEEL 100.0%  31.5%    <SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#0099ff">T</SPAN><SPAN style="color:#cc0000">KK</SPAN><SPAN style="color:#33cc00">LG</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">GA</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">G</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#009900">W</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">G</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">LL</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">IL</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#33cc00">A</SPAN><SPAN style="color:#666666">-------</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#009900">H</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#33cc00">PVLVAV</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0099ff">T</SPAN><SPAN style="color:#33cc00">A</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#33cc00">M</SPAN><SPAN style="color:#0099ff">T</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">IM</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">A</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#33cc00">LM</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#009900">H</SPAN><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#33cc00">VV</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#009900">FF</SPAN><SPAN style="color:#33cc00">GVA</SPAN>
</PRE><p>To make the letters stand out use the <code class="docutils literal notranslate"><span class="pre">-bold</span></code> option:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="ow">in</span> <span class="n">fasta</span> <span class="o">-</span><span class="n">html</span> <span class="n">head</span> <span class="o">-</span><span class="n">bold</span> <span class="o">-</span><span class="n">coloring</span> <span class="nb">any</span> <span class="n">data</span><span class="o">.</span><span class="n">dat</span> <span class="o">&gt;</span> <span class="n">data</span><span class="o">.</span><span class="n">html</span>
</pre></div>
</div>
<p>giving:</p>
<PRE>
Reference sequence (1): EGFR_HUMAN
Identities normalised by aligned length.
Colored by: property

1 EGFR_HUMAN  100.0% 100.0% <STRONG>  </STRONG> <STRONG><SPAN style="color:#009900">F</SPAN><SPAN style="color:#cc0000">KK</SPAN><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">VLG</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#33cc00">GA</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">G</SPAN><SPAN style="color:#0099ff">T</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#009900">Y</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">GL</SPAN><SPAN style="color:#009900">W</SPAN><SPAN style="color:#33cc00">IP</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">G</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#666666">---------</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">IPVAI</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">A</SPAN><SPAN style="color:#0099ff">TS</SPAN><SPAN style="color:#33cc00">P</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#666666">-</SPAN><SPAN style="color:#33cc00">A</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">IL</SPAN><SPAN style="color:#0033ff">DE</SPAN><SPAN style="color:#33cc00">A</SPAN><SPAN style="color:#009900">Y</SPAN><SPAN style="color:#33cc00">VMA</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#33cc00">P</SPAN><SPAN style="color:#009900">H</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#ffff00">C</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#33cc00">LLGI</SPAN><SPAN style="color:#ffff00">C</SPAN></STRONG> <STRONG>  </STRONG>
2 PR2_DROME    97.1%  35.7% <STRONG>  </STRONG> <STRONG><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#33cc00">LG</SPAN><SPAN style="color:#0099ff">T</SPAN><SPAN style="color:#33cc00">G</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">GIV</SPAN><SPAN style="color:#6600cc">QQ</SPAN><SPAN style="color:#33cc00">GV</SPAN><SPAN style="color:#009900">W</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#33cc00">G</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#666666">-----------</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#33cc00">VAI</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#ffff00">C</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#ffff00">C</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#33cc00">M</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#666666">-</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#33cc00">PM</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">AAIM</SPAN><SPAN style="color:#009900">H</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#009900">H</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#33cc00">IV</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#009900">Y</SPAN><SPAN style="color:#33cc00">GVV</SPAN></STRONG> <STRONG>  </STRONG>
3 ITK_HUMAN    90.0%  32.9% <STRONG>  </STRONG> <STRONG><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#0099ff">T</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">IG</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#33cc00">G</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">GLV</SPAN><SPAN style="color:#009900">H</SPAN><SPAN style="color:#33cc00">LG</SPAN><SPAN style="color:#009900">YW</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#666666">--------------</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">VAI</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0099ff">T</SPAN><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">GAM</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#666666">---</SPAN><SPAN style="color:#0033ff">EED</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#0033ff">EE</SPAN><SPAN style="color:#33cc00">A</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">VMM</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#009900">H</SPAN><SPAN style="color:#33cc00">P</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">LV</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#009900">Y</SPAN><SPAN style="color:#33cc00">GV</SPAN><SPAN style="color:#ffff00">C</SPAN></STRONG> <STRONG>  </STRONG>
4 PTK7_HUMAN   97.1%  21.2% <STRONG>  </STRONG> <STRONG><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#33cc00">IGVG</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">GAVVLA</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">M</SPAN><SPAN style="color:#0099ff">T</SPAN><SPAN style="color:#33cc00">GL</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#666666">-X</SPAN><SPAN style="color:#33cc00">LP</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">G</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#33cc00">M</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#33cc00">A</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#33cc00">GVALVAV</SPAN><SPAN style="color:#cc0000">KK</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">P</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#666666">-</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">VL</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">M</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#009900">H</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#33cc00">IV</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#33cc00">LLAI</SPAN><SPAN style="color:#ffff00">C</SPAN></STRONG> <STRONG>  </STRONG>
5 KIN31_CAEEL 100.0%  31.5% <STRONG>  </STRONG> <STRONG><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#0099ff">T</SPAN><SPAN style="color:#cc0000">KK</SPAN><SPAN style="color:#33cc00">LG</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">GA</SPAN><SPAN style="color:#009900">F</SPAN><SPAN style="color:#33cc00">G</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">V</SPAN><SPAN style="color:#009900">W</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">G</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">LL</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">IL</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#33cc00">A</SPAN><SPAN style="color:#666666">-------</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#009900">H</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#33cc00">PVLVAV</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0099ff">T</SPAN><SPAN style="color:#33cc00">A</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#0099ff">S</SPAN><SPAN style="color:#33cc00">M</SPAN><SPAN style="color:#0099ff">T</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#6600cc">Q</SPAN><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">IM</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#0033ff">E</SPAN><SPAN style="color:#33cc00">A</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#33cc00">LM</SPAN><SPAN style="color:#cc0000">R</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#33cc00">L</SPAN><SPAN style="color:#0033ff">D</SPAN><SPAN style="color:#009900">H</SPAN><SPAN style="color:#33cc00">I</SPAN><SPAN style="color:#6600cc">N</SPAN><SPAN style="color:#33cc00">VV</SPAN><SPAN style="color:#cc0000">K</SPAN><SPAN style="color:#009900">FF</SPAN><SPAN style="color:#33cc00">GVA</SPAN></STRONG> <STRONG>  </STRONG>
</PRE><p>Or change the colouring to use blocked letters with <code class="docutils literal notranslate"><span class="pre">-css</span> <span class="pre">on</span></code> instead:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="ow">in</span> <span class="n">fasta</span> <span class="o">-</span><span class="n">html</span> <span class="n">head</span> <span class="o">-</span><span class="n">css</span> <span class="n">on</span> <span class="o">-</span><span class="n">coloring</span> <span class="nb">any</span> <span class="n">data</span><span class="o">.</span><span class="n">dat</span> <span class="o">&gt;</span> <span class="n">data</span><span class="o">.</span><span class="n">html</span>
</pre></div>
</div>
<p>giving:</p>
<PRE>
Reference sequence (1): EGFR_HUMAN
Identities normalised by aligned length.
Colored by: property

                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    <SPAN CLASS=S13>F</SPAN><SPAN CLASS=S18>KK</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>VLG</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>GA</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S13>Y</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>GL</SPAN><SPAN CLASS=S13>W</SPAN><SPAN CLASS=S14>IP</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">---------</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>IPVAI</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S9>TS</SPAN><SPAN CLASS=S14>P</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>IL</SPAN><SPAN CLASS=S12>DE</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S13>Y</SPAN><SPAN CLASS=S14>VMA</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>P</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S7>C</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S14>LLGI</SPAN><SPAN CLASS=S7>C</SPAN>
2 PR2_DROME    97.1%  35.7%    <SPAN CLASS=S14>I</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>LG</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>GIV</SPAN><SPAN CLASS=S8>QQ</SPAN><SPAN CLASS=S14>GV</SPAN><SPAN CLASS=S13>W</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">-----------</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>VAI</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S7>C</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S7>C</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S9>S</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>PM</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>AAIM</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>IV</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S13>Y</SPAN><SPAN CLASS=S14>GVV</SPAN>
3 ITK_HUMAN    90.0%  32.9%    <SPAN CLASS=S14>L</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>IG</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>GLV</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S14>LG</SPAN><SPAN CLASS=S13>YW</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S8>N</SPAN><SPAN style="color:#666666">--------------</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>VAI</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>GAM</SPAN><SPAN CLASS=S9>S</SPAN><SPAN style="color:#666666">---</SPAN><SPAN CLASS=S12>EED</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S12>EE</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>VMM</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S14>P</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>LV</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S13>Y</SPAN><SPAN CLASS=S14>GV</SPAN><SPAN CLASS=S7>C</SPAN>
4 PTK7_HUMAN   97.1%  21.2%    <SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>IGVG</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>GAVVLA</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>GL</SPAN><SPAN CLASS=S9>S</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=T20>X</SPAN><SPAN CLASS=S14>LP</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S14>GVALVAV</SPAN><SPAN CLASS=S18>KK</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>P</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S12>D</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>VL</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>IV</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>LLAI</SPAN><SPAN CLASS=S7>C</SPAN>
5 KIN31_CAEEL 100.0%  31.5%    <SPAN CLASS=S14>V</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S18>KK</SPAN><SPAN CLASS=S14>LG</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>GA</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S13>W</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>LL</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>IL</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#666666">-------</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>PVLVAV</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>IM</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S14>LM</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>VV</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S13>FF</SPAN><SPAN CLASS=S14>GVA</SPAN>
</PRE><p>You can combine <code class="docutils literal notranslate"><span class="pre">-css</span> <span class="pre">on</span></code> with <code class="docutils literal notranslate"><span class="pre">-bold</span></code> to make the blocks and letters even
more prominent.</p>
<p>If your data are DNA or RNA, add the option <code class="docutils literal notranslate"><span class="pre">-moltype</span> <span class="pre">dna</span></code> (or <code class="docutils literal notranslate"><span class="pre">rna</span></code> or
<code class="docutils literal notranslate"><span class="pre">na</span></code> for “nucleic acid”) to change to the default nucleotide
colourmap. Here’s an MView run on some BLASTN data demonstrating some other
options as well:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="ow">in</span> <span class="n">blast</span> <span class="o">-</span><span class="n">html</span> <span class="n">head</span> <span class="o">-</span><span class="n">css</span> <span class="n">on</span> <span class="o">-</span><span class="n">coloring</span> <span class="n">identity</span> <span class="o">-</span><span class="n">moltype</span> <span class="n">dna</span> <span class="o">-</span><span class="n">top</span> <span class="mi">5</span> <span class="o">-</span><span class="nb">range</span> <span class="mi">250</span><span class="p">:</span><span class="mi">310</span> <span class="n">blastn</span><span class="o">.</span><span class="n">dat</span>
</pre></div>
</div>
<p>which (slightly edited to reduce space) produced:</p>
<PRE>
HSP processing: ranked
Query orientation: +

                                              cov    pid   query   sbjct 250 [         .         .         .         .         3         ] 310
  EMBOSS_001         bits E-value  N qy ht 100.0% 100.0%   1:521             <SPAN CLASS=S9>T</SPAN><SPAN CLASS=S12>GAAG</SPAN><SPAN CLASS=S9>CCT</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>C</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CTT</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CTC</SPAN><SPAN CLASS=S12>AGGA</SPAN><SPAN CLASS=S9>CTC</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>TC</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S12>GA</SPAN><SPAN CLASS=S9>CT</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>C</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CC</SPAN><SPAN CLASS=S12>AA</SPAN><SPAN CLASS=S9>TTC</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>TCTT</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CTC</SPAN><SPAN CLASS=S12>AGGA</SPAN><SPAN CLASS=S9>CT</SPAN>
1 EM_EST:GT222018.2  1033     0.0  1  +  + 100.0% 100.0%   1:521   4:524     <SPAN CLASS=S9>T</SPAN><SPAN CLASS=S12>GAAG</SPAN><SPAN CLASS=S9>CCT</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>C</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CTT</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CTC</SPAN><SPAN CLASS=S12>AGGA</SPAN><SPAN CLASS=S9>CTC</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>TC</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S12>GA</SPAN><SPAN CLASS=S9>CT</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>C</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CC</SPAN><SPAN CLASS=S12>AA</SPAN><SPAN CLASS=S9>TTC</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>TCTT</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CTC</SPAN><SPAN CLASS=S12>AGGA</SPAN><SPAN CLASS=S9>CT</SPAN>
2 EM_EST:GT222017.1   186   4e-43  1  +  +  88.5%  98.2% 256:372 205:318     <SPAN style="color:#666666">------</SPAN><SPAN CLASS=S9>CT</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>C</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CTT</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CTC</SPAN><SPAN CLASS=S12>AGGA</SPAN><SPAN CLASS=S9>CTC</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>TC</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S12>GA</SPAN><SPAN CLASS=S9>CT</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>C</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CC</SPAN><SPAN CLASS=S12>AA</SPAN><SPAN CLASS=S9>TTC</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>T</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S9>TT</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CTC</SPAN><SPAN CLASS=S12>AGGA</SPAN><SPAN CLASS=S9>CT</SPAN>
3 EM_EST:GT222024.2   182   7e-42  1  +  +  80.3% 100.0% 262:372  96:209     <SPAN style="color:#666666">------------</SPAN><SPAN CLASS=S9>TT</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CTC</SPAN><SPAN CLASS=S12>AGGA</SPAN><SPAN CLASS=S9>CTC</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>TC</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S12>GA</SPAN><SPAN CLASS=S9>CT</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>C</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CC</SPAN><SPAN CLASS=S12>AA</SPAN><SPAN CLASS=S9>TTC</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>TCtt</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CTC</SPAN><SPAN CLASS=S12>AGGA</SPAN><SPAN CLASS=S9>CT</SPAN>
4 EM_EST:GT222023.2   182   7e-42  1  +  +  80.3% 100.0% 262:372  96:209     <SPAN style="color:#666666">------------</SPAN><SPAN CLASS=S9>TT</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CTC</SPAN><SPAN CLASS=S12>AGGA</SPAN><SPAN CLASS=S9>CTC</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>TC</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S12>GA</SPAN><SPAN CLASS=S9>CT</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>C</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CC</SPAN><SPAN CLASS=S12>AA</SPAN><SPAN CLASS=S9>TTC</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>TCtt</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CTC</SPAN><SPAN CLASS=S12>AGGA</SPAN><SPAN CLASS=S9>CT</SPAN>
5 EM_EST:GT222054.2   178   1e-40  1  +  +  52.5% 100.0% 279:372    4:97     <SPAN style="color:#666666">-----------------------------</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S12>GA</SPAN><SPAN CLASS=S9>CT</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>C</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CC</SPAN><SPAN CLASS=S12>AA</SPAN><SPAN CLASS=S9>TTC</SPAN><SPAN CLASS=S12>G</SPAN><SPAN CLASS=S9>TCTT</SPAN><SPAN CLASS=S12>A</SPAN><SPAN CLASS=S9>CTC</SPAN><SPAN CLASS=S12>AGGA</SPAN><SPAN CLASS=S9>CT</SPAN>
</PRE><p>showing scoring and sequence range information parsed from the BLASTN run, and
using the default nucleotide colouring scheme (purines, dark blue;
pyrimidines, light blue). Notice the lower-cased pairs of thymines near the
end of sequences 3 and 4, columns 299–300 indicating where a segment of hit
sequence has been excised to close a gap in the query (see
<a class="reference internal" href="faq.html#ref-funny-sequences"><span class="std std-ref">Why are some sequence symbols lowercased?</span></a>).</p>
</section>
<section id="controlling-the-amount-of-html">
<h3>Controlling the amount of HTML<a class="headerlink" href="#controlling-the-amount-of-html" title="Permalink to this headline">¶</a></h3>
<p>There are several values that can be passed to the <code class="docutils literal notranslate"><span class="pre">-html</span></code> option: <code class="docutils literal notranslate"><span class="pre">head</span></code>,
<code class="docutils literal notranslate"><span class="pre">body</span></code>, <code class="docutils literal notranslate"><span class="pre">data</span></code>, <code class="docutils literal notranslate"><span class="pre">full</span></code>, <code class="docutils literal notranslate"><span class="pre">off</span></code>.</p>
<p><strong>Mode</strong> <code class="docutils literal notranslate"><span class="pre">head</span></code></p>
<p>Produces a complete web page. Output includes the style sheet if <code class="docutils literal notranslate"><span class="pre">-css</span> <span class="pre">on</span></code>
was given. The most common situation.</p>
<p><strong>Mode</strong> <code class="docutils literal notranslate"><span class="pre">body</span></code></p>
<p>Produces just the <code class="docutils literal notranslate"><span class="pre">&lt;BODY&gt;&lt;/BODY&gt;</span></code> part of the web page.  Note: the style
sheet produced by <code class="docutils literal notranslate"><span class="pre">-css</span> <span class="pre">on</span></code> will be missing.</p>
<p><strong>Mode</strong> <code class="docutils literal notranslate"><span class="pre">data</span></code></p>
<p>Produces just the alignment part of the web page. Note: any style sheet
produced by <code class="docutils literal notranslate"><span class="pre">-css</span> <span class="pre">on</span></code> will be missing.</p>
<p><strong>Mode</strong> <code class="docutils literal notranslate"><span class="pre">full</span></code></p>
<p>Produces a complete web page with the <code class="docutils literal notranslate"><span class="pre">MIME-type</span> <span class="pre">&quot;text/html&quot;</span></code>, suitable for
serving directly from a web server. Output includes the style sheet if <code class="docutils literal notranslate"><span class="pre">-css</span>
<span class="pre">on</span></code> was given.</p>
<p><strong>Mode</strong> <code class="docutils literal notranslate"><span class="pre">off</span></code></p>
<p>Switches off HTML (default).</p>
</section>
<section id="using-an-external-css-style-sheet">
<h3>Using an external CSS style sheet<a class="headerlink" href="#using-an-external-css-style-sheet" title="Permalink to this headline">¶</a></h3>
<p>The option <code class="docutils literal notranslate"><span class="pre">-listcss</span></code> dumps the style sheet to stdout, so you can share that
across MView invocations from a web server. Each would be of the form:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="n">css</span> <span class="n">URL</span>
</pre></div>
</div>
<p>where the URL specifies the location of the style sheet as seen by the web
server (i.e., <code class="docutils literal notranslate"><span class="pre">file:///some/path</span></code> or <code class="docutils literal notranslate"><span class="pre">http://server/path</span></code>).</p>
<p>If you build a new colourmap you can load it into MView and save the new CSS
file. Suppose you have a new colourmap in <code class="docutils literal notranslate"><span class="pre">newcolmap.dat</span></code>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="n">colorfile</span> <span class="n">newcolmap</span><span class="o">.</span><span class="n">dat</span> <span class="o">-</span><span class="n">listcss</span>
</pre></div>
</div>
<p>will dump the new style sheet for use as before.</p>
</section>
</section>
<section id="consensus-sequences">
<span id="ref-consensus-sequences"></span><h2>Consensus sequences<a class="headerlink" href="#consensus-sequences" title="Permalink to this headline">¶</a></h2>
<section id="clustal-conservation-line">
<h3>Clustal conservation line<a class="headerlink" href="#clustal-conservation-line" title="Permalink to this headline">¶</a></h3>
<p>A Clustal-style conservation line of <code class="docutils literal notranslate"><span class="pre">*:.</span></code> symbols can be added to any
alignment (not just one from CLUSTAL itself) using the <code class="docutils literal notranslate"><span class="pre">-conservation</span> <span class="pre">on</span></code>
option:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">conservation</span> <span class="n">on</span>
</pre></div>
</div>
<p>giving:</p>
<PRE>
                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIPVAIKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC
2 PR2_DROME    97.1%  35.7%    ISVNKQLGTGEFGIVQQGVWSNGNE-----------RIQVAIKCLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV
3 ITK_HUMAN    90.0%  32.9%    LTFVQEIGSGQFGLVHLGYWLN--------------KDKVAIKTIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC
4 PTK7_HUMAN   97.1%  21.2%    IREVKQIGVGQFGAVVLAEMTGLS-XLPKGSMNADGVALVAVKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC
5 KIN31_CAEEL 100.0%  31.5%    VELTKKLGEGAFGEVWKGKLLKILDA-------NHQPVLVAVKTAKLESMTKEQIKEIMREARLMRNLDHINVVKFFGVA
  clustal                      .   : :* * ** *  .                     **:*             .:  *  .* .:.:  : :: .:

</PRE><p>The symbols are <code class="docutils literal notranslate"><span class="pre">*</span></code> for full column identity, and <code class="docutils literal notranslate"><span class="pre">:</span></code> or <code class="docutils literal notranslate"><span class="pre">.</span></code> for strong
and weak amino acid grouping, respectively, as defined in CLUSTAL.</p>
<p>For DNA or RNA sequences, if the molecule type was set to nucleic acid with
<code class="docutils literal notranslate"><span class="pre">-moltype</span> <span class="pre">na</span></code> or <code class="docutils literal notranslate"><span class="pre">dna</span></code> or <code class="docutils literal notranslate"><span class="pre">rna</span></code>, then the clustal conservation line will
show only the column identities.</p>
<p>Note: these conservation lines can be generated for any subset of rows
extracted using the various row filtering options (see
<a class="reference internal" href="#ref-filtering-rows"><span class="std std-ref">Filtering rows</span></a>).</p>
</section>
<section id="consensus-lines">
<h3>Consensus lines<a class="headerlink" href="#consensus-lines" title="Permalink to this headline">¶</a></h3>
<p>Consensus lines can be added beneath the alignment using <code class="docutils literal notranslate"><span class="pre">-consensus</span> <span class="pre">on</span></code>. By
default, this adds four extra lines of consensus sequences computed at various
thresholds of percentage composition of the columns.</p>
<p>There are default consensus patterns for protein and nucleotide (either DNA or
RNA) sequences. MView starts up with the default protein consensus pattern,
for example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">consensus</span> <span class="n">on</span>
</pre></div>
</div>
<p>gives:</p>
<PRE>
                    cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN     100.0% 100.0%    FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIPVAIKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC
2 PR2_DROME       97.1%  35.7%    ISVNKQLGTGEFGIVQQGVWSNGNE-----------RIQVAIKCLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV
3 ITK_HUMAN       90.0%  32.9%    LTFVQEIGSGQFGLVHLGYWLN--------------KDKVAIKTIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC
4 PTK7_HUMAN      97.1%  21.2%    IREVKQIGVGQFGAVVLAEMTGLS-XLPKGSMNADGVALVAVKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC
5 KIN31_CAEEL    100.0%  31.5%    VELTKKLGEGAFGEVWKGKLLKILDA-------NHQPVLVAVKTAKLESMTKEQIKEIMREARLMRNLDHINVVKFFGVA
  consensus/100%                  hp..p.lG.GtFG.V..u.h...................VAlKphp.t........ph.cEh.hM.plpp.plsphhuls
  consensus/90%                   hp..p.lG.GtFG.V..u.h...................VAlKphp.t........ph.cEh.hM.plpp.plsphhuls
  consensus/80%                   lphsKplGsGtFGhVhhGhhhs..............hh.VAlKpl+.ts.s....p-hhcEAtlMtplpH.plVpLhGls
  consensus/70%                   lphsKplGsGtFGhVhhGhhhs..............hh.VAlKpl+.ts.s....p-hhcEAtlMtplpH.plVpLhGls
</PRE></section>
<section id="changing-consensus-thresholds">
<h3>Changing consensus thresholds<a class="headerlink" href="#changing-consensus-thresholds" title="Permalink to this headline">¶</a></h3>
<p>The default consensus mechanism displays consensus lines calculated at four
levels of identity (100%, 90%, 80%, 70%). This can be changed to show as many
or as few consensus lines at any level of percent identity between 50 and 100%
using the <code class="docutils literal notranslate"><span class="pre">-con_threshold</span></code> option and a comma-separated list of identities:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">consensus</span> <span class="n">on</span> <span class="o">-</span><span class="n">con_threshold</span> <span class="mi">80</span>
</pre></div>
</div>
<p>would give a single consensus line calculated at 80% identity, while:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">consensus</span> <span class="n">on</span> <span class="o">-</span><span class="n">con_threshold</span> <span class="mi">80</span><span class="p">,</span><span class="mi">65</span>
</pre></div>
</div>
<p>would produce two lines at 80% and 65% identity.</p>
</section>
<section id="consensus-pattern-definitions">
<h3>Consensus pattern definitions<a class="headerlink" href="#consensus-pattern-definitions" title="Permalink to this headline">¶</a></h3>
<p>Consensus patterns are based on equivalence classes, that is, sets of residues
that share some predefined property. These classes are not mutually exclusive
and the consensus mechanism will choose the most specific class that
summarizes a given column at the desired percent identity.</p>
<p>The default for protein alignments is called <code class="docutils literal notranslate"><span class="pre">P1</span></code> and is defined by
physicochemical property as follows:</p>
<PRE>
<SPAN style="color:#000000">[P1]</SPAN>
<SPAN style="color:#aa6666">#Protein consensus: conserved physicochemical classes, derived from
#the Venn diagrams of: Taylor W. R. (1986). The classification of amino acid
#conservation. J. Theor. Biol. 119:205-218.
#description =>  symbol  members</SPAN>
.            =>  .
A            =>  A       { A }
C            =>  C       { C }
D            =>  D       { D }
E            =>  E       { E }
F            =>  F       { F }
G            =>  G       { G }
H            =>  H       { H }
I            =>  I       { I }
K            =>  K       { K }
L            =>  L       { L }
M            =>  M       { M }
N            =>  N       { N }
P            =>  P       { P }
Q            =>  Q       { Q }
R            =>  R       { R }
S            =>  S       { S }
T            =>  T       { T }
V            =>  V       { V }
W            =>  W       { W }
Y            =>  Y       { Y }
alcohol      =>  o       { S, T }
aliphatic    =>  l       { I, L, V }
aromatic     =>  a       { F, H, W, Y }
charged      =>  c       { D, E, H, K, R }
hydrophobic  =>  h       { A, C, F, G, H, I, K, L, M, R, T, V, W, Y }
negative     =>  -       { D, E }
polar        =>  p       { C, D, E, H, K, N, Q, R, S, T }
positive     =>  +       { H, K, R }
small        =>  s       { A, C, D, G, N, P, S, T, V }
tiny         =>  u       { A, G, S }
turnlike     =>  t       { A, C, D, E, G, H, K, N, Q, R, S, T }
stop         =>  *       { * }
</PRE><p>The default nucleotide consensus pattern is <code class="docutils literal notranslate"><span class="pre">D1</span></code> grouping bases by ring type
(purine, pyrimidine). It is selected when any of the nucleotide molecule types
is set <code class="docutils literal notranslate"><span class="pre">-moltype</span> <span class="pre">na</span></code> (for “nucleic acid”; also <code class="docutils literal notranslate"><span class="pre">dna</span></code> or <code class="docutils literal notranslate"><span class="pre">rna</span></code>), for
example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">consensus</span> <span class="n">on</span> <span class="o">-</span><span class="n">moltype</span> <span class="n">dna</span>
</pre></div>
</div>
<p>and has the following definition:</p>
<PRE>
<SPAN style="color:#000000">[D1]</SPAN>
<SPAN style="color:#aa6666">#DNA consensus: conserved ring types
#Ambiguous base R is purine: A or G
#Ambiguous base Y is pyrimidine: C or T or U
#description =>  symbol  members</SPAN>
.            =>  .
A            =>  A       { A }
C            =>  C       { C }
G            =>  G       { G }
T            =>  T       { T }
U            =>  U       { U }
purine       =>  r       { A, G, R }
pyrimidine   =>  y       { C, T, U, Y }
</PRE></section>
<section id="changing-consensus-patterns">
<span id="ref-changeing-consensus-patterns"></span><h3>Changing consensus patterns<a class="headerlink" href="#changing-consensus-patterns" title="Permalink to this headline">¶</a></h3>
<p>The available list of built-in patterns can be seen with <code class="docutils literal notranslate"><span class="pre">-listgroups</span></code>.</p>
<p>Alternative equivalence classes can be selected using <code class="docutils literal notranslate"><span class="pre">-con_groupmap</span></code>. For
example, to select the <code class="docutils literal notranslate"><span class="pre">CYS</span></code> built-in consensus pattern to show only
conserved cysteines you would use an invocation like:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">consensus</span> <span class="n">on</span> <span class="o">-</span><span class="n">con_groupmap</span> <span class="n">CYS</span>
</pre></div>
</div>
<p>New groups can be defined in the same format and read in from a file using
the <code class="docutils literal notranslate"><span class="pre">-groupfile</span></code> option.</p>
</section>
<section id="showing-conserved-residues-or-conserved-classes">
<span id="ref-conserved-symbols-or-conserved-classes"></span><h3>Showing conserved residues or conserved classes<a class="headerlink" href="#showing-conserved-residues-or-conserved-classes" title="Permalink to this headline">¶</a></h3>
<p>Two options <code class="docutils literal notranslate"><span class="pre">-con_ignore</span></code> and <code class="docutils literal notranslate"><span class="pre">-con_gaps</span></code> can be used to tune the
consensus lines. Consider the following alignment:</p>
<PRE>
                    cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN     100.0% 100.0%    FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIPVAIKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC
2 PR2_DROME       97.1%  35.7%    ISVNKQLGTGEFGIVQQGVWSNGNE-----------RIQVAIKCLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV
3 ITK_HUMAN       90.0%  32.9%    LTFVQEIGSGQFGLVHLGYWLN--------------KDKVAIKTIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC
4 PTK7_HUMAN      97.1%  21.2%    IREVKQIGVGQFGAVVLAEMTGLS-XLPKGSMNADGVALVAVKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC
5 KIN31_CAEEL    100.0%  31.5%    VELTKKLGEGAFGEVWKGKLLKILDA-------NHQPVLVAVKTAKLESMTKEQIKEIMREARLMRNLDHINVVKFFGVA
</PRE><p>The default consensus pattern for proteins, with these options:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">consensus</span> <span class="n">on</span> <span class="o">-</span><span class="n">con_threshold</span> <span class="mi">80</span>
</pre></div>
</div>
<p>would add this consensus line:</p>
<PRE>
  consensus/80%                  lphsKplGsGtFGhVhhGhhhs..............hh.VAlKpl+.ts.s....p-hhcEAtlMtplpH.plVpLhGls
</PRE><p>comprising a mixture of conserved residue classes and residues, whichever is
more specific.</p>
<p>If you just want to see the conserved physicochemical classes, use <code class="docutils literal notranslate"><span class="pre">-con_ignore</span> <span class="pre">singleton</span></code>:</p>
<PRE>
  consensus/80%                  lphs+plusutauhlhhuhhhs..............hh.lul+pl+.ts.s....p-hhc-utlhtplp+.pllplhuls
</PRE><p>Alternatively, to see just the conserved residues, use <code class="docutils literal notranslate"><span class="pre">-con_ignore</span> <span class="pre">class</span></code>:</p>
<PRE>
  consensus/80%                  ....K..G.G.FG.V..G.....................VA.K.................EA..M....H...V.L.G..
</PRE><p>Lastly, the default consensus computation counts gap characters in each
column, so that gapped regions are diluted and may not show up in the
consensus. Building on the last example, setting <code class="docutils literal notranslate"><span class="pre">-con_gaps</span> <span class="pre">off</span></code> prevents
this:</p>
<PRE>
  consensus/80%                  ....K..G.G.FG.V..G........LPKGSMN......VA.K.........E.......EA..M....H...V.L.G..
</PRE><p>The consensus sequence now runs the full length of the alignment because the
insert in sequence 4 spanning the gap has been added to the consensus. This is
a little contrived in this case, but is sometimes useful when you want to
preserve as much of the alignment as possible.</p>
<p>These options work similarly with nucleotide alignments and with any other
consensus pattern you choose.</p>
<p>Note: it is possible to colour the consensus sequences independently of the
alignment (see <a class="reference internal" href="#ref-consensus-colouring"><span class="std std-ref">Consensus pattern colouring</span></a>).</p>
</section>
</section>
<section id="colouring-modes">
<h2>Colouring modes<a class="headerlink" href="#colouring-modes" title="Permalink to this headline">¶</a></h2>
<section id="alignment-colouring">
<span id="ref-alignment-colouring"></span><h3>Alignment colouring<a class="headerlink" href="#alignment-colouring" title="Permalink to this headline">¶</a></h3>
<p>There are several basic ways to colour the alignment using the <code class="docutils literal notranslate"><span class="pre">-coloring</span></code>
option which takes five modes: <code class="docutils literal notranslate"><span class="pre">any</span></code>, <code class="docutils literal notranslate"><span class="pre">identity</span></code>, <code class="docutils literal notranslate"><span class="pre">mismatch</span></code>,
<code class="docutils literal notranslate"><span class="pre">consensus</span></code>, <code class="docutils literal notranslate"><span class="pre">group</span></code>. These all have default associated colour schemes,
but you can supply a different one or just a single colour by name (see the
description for the <code class="docutils literal notranslate"><span class="pre">mismatch</span></code> mode for an example).</p>
<p><strong>Mode</strong> <code class="docutils literal notranslate"><span class="pre">any</span></code></p>
<p>The simplest is to colour every residue according to the currently selected
colourmap:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="n">html</span> <span class="n">head</span> <span class="o">-</span><span class="n">css</span> <span class="n">on</span> <span class="o">-</span><span class="n">coloring</span> <span class="nb">any</span>
</pre></div>
</div>
<p>gives:</p>
<PRE>
Colored by: property

                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    <SPAN CLASS=S13>F</SPAN><SPAN CLASS=S18>KK</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>VLG</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>GA</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S13>Y</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>GL</SPAN><SPAN CLASS=S13>W</SPAN><SPAN CLASS=S14>IP</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">---------</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>IPVAI</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S9>TS</SPAN><SPAN CLASS=S14>P</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>IL</SPAN><SPAN CLASS=S12>DE</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S13>Y</SPAN><SPAN CLASS=S14>VMA</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>P</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S7>C</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S14>LLGI</SPAN><SPAN CLASS=S7>C</SPAN>
2 PR2_DROME    97.1%  35.7%    <SPAN CLASS=S14>I</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>LG</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>GIV</SPAN><SPAN CLASS=S8>QQ</SPAN><SPAN CLASS=S14>GV</SPAN><SPAN CLASS=S13>W</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">-----------</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>VAI</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S7>C</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S7>C</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S9>S</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>PM</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>AAIM</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>IV</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S13>Y</SPAN><SPAN CLASS=S14>GVV</SPAN>
3 ITK_HUMAN    90.0%  32.9%    <SPAN CLASS=S14>L</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>IG</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>GLV</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S14>LG</SPAN><SPAN CLASS=S13>YW</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S8>N</SPAN><SPAN style="color:#666666">--------------</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>VAI</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>GAM</SPAN><SPAN CLASS=S9>S</SPAN><SPAN style="color:#666666">---</SPAN><SPAN CLASS=S12>EED</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S12>EE</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>VMM</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S14>P</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>LV</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S13>Y</SPAN><SPAN CLASS=S14>GV</SPAN><SPAN CLASS=S7>C</SPAN>
4 PTK7_HUMAN   97.1%  21.2%    <SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>IGVG</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>GAVVLA</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>GL</SPAN><SPAN CLASS=S9>S</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=T20>X</SPAN><SPAN CLASS=S14>LP</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S14>GVALVAV</SPAN><SPAN CLASS=S18>KK</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>P</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S12>D</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>VL</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>IV</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>LLAI</SPAN><SPAN CLASS=S7>C</SPAN>
5 KIN31_CAEEL 100.0%  31.5%    <SPAN CLASS=S14>V</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S18>KK</SPAN><SPAN CLASS=S14>LG</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>GA</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S13>W</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>LL</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>IL</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#666666">-------</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>PVLVAV</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>IM</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S14>LM</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>VV</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S13>FF</SPAN><SPAN CLASS=S14>GVA</SPAN>
</PRE><p><strong>Mode</strong> <code class="docutils literal notranslate"><span class="pre">identity</span></code></p>
<p>You can colour only those residues that are identical to some reference
sequence (usually the query or first row) with:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">coloring</span> <span class="n">identity</span>
</pre></div>
</div>
<p>to produce:</p>
<PRE>
Colored by: identity

                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    <SPAN CLASS=S13>F</SPAN><SPAN CLASS=S18>KK</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>VLG</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>GA</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S13>Y</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>GL</SPAN><SPAN CLASS=S13>W</SPAN><SPAN CLASS=S14>IP</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">---------</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>IPVAI</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S9>TS</SPAN><SPAN CLASS=S14>P</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>IL</SPAN><SPAN CLASS=S12>DE</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S13>Y</SPAN><SPAN CLASS=S14>VMA</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>P</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S7>C</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S14>LLGI</SPAN><SPAN CLASS=S7>C</SPAN>
2 PR2_DROME    97.1%  35.7%    <SPAN style="color:#666666">ISVN</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">Q</SPAN><SPAN CLASS=S14>LG</SPAN><SPAN style="color:#666666">T</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">QQ</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S13>W</SPAN><SPAN style="color:#666666">SNGN</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">-----------R</SPAN><SPAN CLASS=S14>I</SPAN><SPAN style="color:#666666">Q</SPAN><SPAN CLASS=S14>VAI</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">C</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">CRERMQS-NPM</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">F</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#666666">AI</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">H</SPAN><SPAN CLASS=S9>S</SPAN><SPAN style="color:#666666">IEHENIV</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">Y</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">VV</SPAN>
3 ITK_HUMAN    90.0%  32.9%    <SPAN style="color:#666666">LTFVQEI</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">Q</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">HL</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">Y</SPAN><SPAN CLASS=S13>W</SPAN><SPAN style="color:#666666">LN--------------</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">DK</SPAN><SPAN CLASS=S14>VAI</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">TI</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">GAMS---EEDFIE</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>VM</SPAN><SPAN style="color:#666666">MKLSH</SPAN><SPAN CLASS=S14>P</SPAN><SPAN style="color:#666666">KLVQ</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">Y</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S7>C</SPAN>
4 PTK7_HUMAN   97.1%  21.2%    <SPAN style="color:#666666">IREV</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">QI</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">Q</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">A</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">VLAEMTGLS-XLPKGSMNADGVAL</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">KPDV</SPAN><SPAN CLASS=S9>S</SPAN><SPAN style="color:#666666">D-EVLQSFDK</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">IKF</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">SQLQHDSIVQ</SPAN><SPAN CLASS=S14>LL</SPAN><SPAN style="color:#666666">A</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S7>C</SPAN>
5 KIN31_CAEEL 100.0%  31.5%    <SPAN style="color:#666666">VELT</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S14>LG</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>GA</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">W</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">KLLKILDA-------NHQPVL</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">TAKLESMT</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">EQI</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>I</SPAN><SPAN style="color:#666666">MR</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#666666">RL</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">RNL</SPAN><SPAN CLASS=S12>D</SPAN><SPAN style="color:#666666">HIN</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">VKFF</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">VA</SPAN>
</PRE><p>or with respect to another row (let’s use row 4):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">coloring</span> <span class="n">identity</span> <span class="o">-</span><span class="n">ref</span> <span class="mi">4</span>
</pre></div>
</div>
<p>giving:</p>
<PRE>
Colored by: identity

                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN   87.2%  21.2%    <SPAN style="color:#666666">FKKI</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">VL</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">A</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">T</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">YKGLWIPEGEK---------VKIP</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">REAT</SPAN><SPAN CLASS=S9>S</SPAN><SPAN style="color:#666666">PK-ANKEILD</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">AYV</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">ASVDNPHVCR</SPAN><SPAN CLASS=S14>LL</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S7>C</SPAN>
2 PR2_DROME    84.6%  25.0%    <SPAN CLASS=S14>I</SPAN><SPAN style="color:#666666">SVN</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">T</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">QQGVWSNGNE-----------RIQ</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">C</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">CRERMQS-NPME</SPAN><SPAN CLASS=S13>F</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">AAI</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">HSIE</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">EN</SPAN><SPAN CLASS=S14>IV</SPAN><SPAN style="color:#666666">R</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">YGVV</SPAN>
3 ITK_HUMAN    80.8%  26.9%    <SPAN style="color:#666666">LTF</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">QE</SPAN><SPAN CLASS=S14>IG</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">H</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">GYWLN--------------KDK</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">TIREGAMS---EED</SPAN><SPAN CLASS=S13>F</SPAN><SPAN style="color:#666666">IE</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">AEV</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">MK</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">PKL</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">YGV</SPAN><SPAN CLASS=S7>C</SPAN>
4 PTK7_HUMAN  100.0% 100.0%    <SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>IGVG</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>GAVVLA</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>GL</SPAN><SPAN CLASS=S9>S</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=T20>X</SPAN><SPAN CLASS=S14>LP</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S14>GVALVAV</SPAN><SPAN CLASS=S18>KK</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>P</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S12>D</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>VL</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>IV</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>LLAI</SPAN><SPAN CLASS=S7>C</SPAN>
5 KIN31_CAEEL  91.0%  22.5%    <SPAN style="color:#666666">VELT</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">KL</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">A</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">WKGKLLKILDA-------NHQPV</SPAN><SPAN CLASS=S14>LVAV</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">TA</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">LESMTK</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">QIKEIMR</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">ARL</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">RN</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">D</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">INV</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">KFFGVA</SPAN>
</PRE><p>in which also you see that the percent identity calculations have been
recomputed with respect to the new row of interest.</p>
<p><strong>Mode</strong> <code class="docutils literal notranslate"><span class="pre">mismatch</span></code></p>
<p>Behaves like <code class="docutils literal notranslate"><span class="pre">identity</span></code> mode, but colours only those residues that differ
from the reference sequence (the query or first row unless specified
otherwise) with:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">coloring</span> <span class="n">mismatch</span>
</pre></div>
</div>
<p>to produce:</p>
<PRE>
Colored by: mismatch

                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    <SPAN style="color:#666666">FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIPVAIKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC</SPAN>
2 PR2_DROME    97.1%  35.7%    <SPAN CLASS=S14>I</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S8>N</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN style="color:#666666">LG</SPAN><SPAN CLASS=S9>T</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">FG</SPAN><SPAN CLASS=S14>I</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S8>QQ</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">W</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S8>N</SPAN><SPAN style="color:#666666">E-----------</SPAN><SPAN CLASS=S18>R</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN style="color:#666666">VAIK</SPAN><SPAN CLASS=S7>C</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S7>C</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S9>S</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>PM</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S13>F</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">EA</SPAN><SPAN CLASS=S14>AI</SPAN><SPAN style="color:#666666">M</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>IV</SPAN><SPAN style="color:#666666">RL</SPAN><SPAN CLASS=S13>Y</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S14>VV</SPAN>
3 ITK_HUMAN    90.0%  32.9%    <SPAN CLASS=S14>L</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>I</SPAN><SPAN style="color:#666666">GSG</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN style="color:#666666">FG</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S13>Y</SPAN><SPAN style="color:#666666">W</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S8>N</SPAN><SPAN style="color:#666666">--------------K</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">VAIK</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>I</SPAN><SPAN style="color:#666666">RE</SPAN><SPAN CLASS=S14>GAM</SPAN><SPAN CLASS=S9>S</SPAN><SPAN style="color:#666666">---</SPAN><SPAN CLASS=S12>EED</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">EA</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">VM</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">P</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>LV</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S13>Y</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">C</SPAN>
4 PTK7_HUMAN   97.1%  21.2%    <SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>I</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN style="color:#666666">FG</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S14>VLA</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>GL</SPAN><SPAN CLASS=S9>S</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=T20>X</SPAN><SPAN CLASS=S14>LP</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S14>GVAL</SPAN><SPAN style="color:#666666">VA</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>P</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=S12>D</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>VL</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S13>F</SPAN><SPAN style="color:#666666">M</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>IV</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN style="color:#666666">LL</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#666666">IC</SPAN>
5 KIN31_CAEEL 100.0%  31.5%    <SPAN CLASS=S14>V</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S9>T</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">LG</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">GAFG</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S13>W</SPAN><SPAN style="color:#666666">KG</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>LL</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>IL</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#666666">-------</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>PVL</SPAN><SPAN style="color:#666666">VA</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S9>T</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>I</SPAN><SPAN style="color:#666666">KEI</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S18>R</SPAN><SPAN style="color:#666666">EA</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">M</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">D</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S8>N</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S13>FF</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S14>VA</SPAN>
</PRE><p><strong>Using a single colour</strong></p>
<p>That’s using the default protein colourmap and rather difficult to see, so
let’s mark all mismatched residues in red:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">coloring</span> <span class="n">mismatch</span> <span class="o">-</span><span class="n">colormap</span> <span class="n">red</span>
</pre></div>
</div>
<p>to produce:</p>
<PRE>
Colored by: mismatch

                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    <SPAN style="color:#666666">FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIPVAIKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC</SPAN>
2 PR2_DROME    97.1%  35.7%    <SPAN CLASS=S2>ISVN</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S2>Q</SPAN><SPAN style="color:#666666">LG</SPAN><SPAN CLASS=S2>T</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S2>E</SPAN><SPAN style="color:#666666">FG</SPAN><SPAN CLASS=S2>I</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S2>QQ</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S2>V</SPAN><SPAN style="color:#666666">W</SPAN><SPAN CLASS=S2>SNGN</SPAN><SPAN style="color:#666666">E-----------</SPAN><SPAN CLASS=S2>R</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S2>Q</SPAN><SPAN style="color:#666666">VAIK</SPAN><SPAN CLASS=S2>C</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S2>CRERMQS</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S2>NPM</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S2>F</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S2>K</SPAN><SPAN style="color:#666666">EA</SPAN><SPAN CLASS=S2>AI</SPAN><SPAN style="color:#666666">M</SPAN><SPAN CLASS=S2>H</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=S2>IEHENIV</SPAN><SPAN style="color:#666666">RL</SPAN><SPAN CLASS=S2>Y</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S2>VV</SPAN>
3 ITK_HUMAN    90.0%  32.9%    <SPAN CLASS=S2>LTFVQEI</SPAN><SPAN style="color:#666666">GSG</SPAN><SPAN CLASS=S2>Q</SPAN><SPAN style="color:#666666">FG</SPAN><SPAN CLASS=S2>L</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S2>HL</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S2>Y</SPAN><SPAN style="color:#666666">W</SPAN><SPAN CLASS=S2>LN</SPAN><SPAN style="color:#666666">--------------K</SPAN><SPAN CLASS=S2>DK</SPAN><SPAN style="color:#666666">VAIK</SPAN><SPAN CLASS=S2>TI</SPAN><SPAN style="color:#666666">RE</SPAN><SPAN CLASS=S2>GAMS</SPAN><SPAN style="color:#666666">---</SPAN><SPAN CLASS=S2>EEDFIE</SPAN><SPAN style="color:#666666">EA</SPAN><SPAN CLASS=S2>E</SPAN><SPAN style="color:#666666">VM</SPAN><SPAN CLASS=S2>MKLSH</SPAN><SPAN style="color:#666666">P</SPAN><SPAN CLASS=S2>KLVQ</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S2>Y</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S2>V</SPAN><SPAN style="color:#666666">C</SPAN>
4 PTK7_HUMAN   97.1%  21.2%    <SPAN CLASS=S2>IREV</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S2>QI</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S2>V</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S2>Q</SPAN><SPAN style="color:#666666">FG</SPAN><SPAN CLASS=S2>A</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S2>VLAEMTGLS</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S2>XLPKGSMNADGVAL</SPAN><SPAN style="color:#666666">VA</SPAN><SPAN CLASS=S2>V</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S2>K</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S2>KPDV</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=S2>D</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S2>EVLQSFDK</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S2>IKF</SPAN><SPAN style="color:#666666">M</SPAN><SPAN CLASS=S2>SQLQHDSIVQ</SPAN><SPAN style="color:#666666">LL</SPAN><SPAN CLASS=S2>A</SPAN><SPAN style="color:#666666">IC</SPAN>
5 KIN31_CAEEL 100.0%  31.5%    <SPAN CLASS=S2>VELT</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S2>K</SPAN><SPAN style="color:#666666">LG</SPAN><SPAN CLASS=S2>E</SPAN><SPAN style="color:#666666">GAFG</SPAN><SPAN CLASS=S2>E</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S2>W</SPAN><SPAN style="color:#666666">KG</SPAN><SPAN CLASS=S2>KLLKILDA</SPAN><SPAN style="color:#666666">-------</SPAN><SPAN CLASS=S2>NHQPVL</SPAN><SPAN style="color:#666666">VA</SPAN><SPAN CLASS=S2>V</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S2>TAKLESMT</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S2>EQI</SPAN><SPAN style="color:#666666">KEI</SPAN><SPAN CLASS=S2>MR</SPAN><SPAN style="color:#666666">EA</SPAN><SPAN CLASS=S2>RL</SPAN><SPAN style="color:#666666">M</SPAN><SPAN CLASS=S2>RNL</SPAN><SPAN style="color:#666666">D</SPAN><SPAN CLASS=S2>HIN</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S2>VKFF</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S2>VA</SPAN>
</PRE><p>This approach of using a single colour works with any of the colouring modes
listed above. To see colours and colormaps use <code class="docutils literal notranslate"><span class="pre">mview</span> <span class="pre">-listcolors</span></code>.</p>
<p>You can add new colours or build your own multicoloured specialist
colourmap. For example, you might want certain important mismatched residues
showing up in one colour against a ground colour of all the other mismatches
(see <a class="reference internal" href="#ref-colourmaps"><span class="std std-ref">Colours</span></a>).</p>
<p><strong>Mode</strong> <code class="docutils literal notranslate"><span class="pre">consensus</span></code></p>
<p>This mode uses the currently selected alignment colourmap to colour only those
residues assigned to a consensus class for each column (see
<a class="reference internal" href="#ref-consensus-sequences"><span class="std std-ref">Consensus sequences</span></a>). The consensus threshold defaults to 70% and
and may be set to another value with the <code class="docutils literal notranslate"><span class="pre">-threshold</span></code> option. In the
following example we add a single consensus line and set the same threshold
for both consensus calculations (they are independent) to 90%:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">coloring</span> <span class="n">consensus</span> <span class="o">-</span><span class="n">threshold</span> <span class="mi">90</span>  <span class="o">-</span><span class="n">consensus</span> <span class="n">on</span> <span class="o">-</span><span class="n">con_threshold</span> <span class="mi">90</span>
</pre></div>
</div>
<p>gives:</p>
<PRE>
Colored by: consensus/90%

                   cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN    100.0% 100.0%    <SPAN CLASS=S13>F</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">KI</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S14>LG</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=S14>GA</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">T</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">YK</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S13>W</SPAN><SPAN style="color:#666666">IPEGEK---------VKIP</SPAN><SPAN CLASS=S14>VAI</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S18>R</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#666666">TSPK-ANK</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>I</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S12>DE</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#666666">Y</SPAN><SPAN CLASS=S14>VM</SPAN><SPAN style="color:#666666">A</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S8>N</SPAN><SPAN style="color:#666666">P</SPAN><SPAN CLASS=S13>H</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=S7>C</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S14>LLGI</SPAN><SPAN CLASS=S7>C</SPAN>
2 PR2_DROME      97.1%  35.7%    <SPAN CLASS=S14>I</SPAN><SPAN CLASS=S9>S</SPAN><SPAN style="color:#666666">VN</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">Q</SPAN><SPAN CLASS=S14>LG</SPAN><SPAN style="color:#666666">T</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">QQ</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S13>W</SPAN><SPAN style="color:#666666">SNGNE-----------RIQ</SPAN><SPAN CLASS=S14>VAI</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S7>C</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S7>C</SPAN><SPAN style="color:#666666">R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">RMQS-NPM</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S13>F</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#666666">A</SPAN><SPAN CLASS=S14>IM</SPAN><SPAN style="color:#666666">H</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>IV</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S13>Y</SPAN><SPAN CLASS=S14>GVV</SPAN>
3 ITK_HUMAN      90.0%  32.9%    <SPAN CLASS=S14>L</SPAN><SPAN CLASS=S9>T</SPAN><SPAN style="color:#666666">FV</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>IG</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">HL</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">Y</SPAN><SPAN CLASS=S13>W</SPAN><SPAN style="color:#666666">LN--------------KDK</SPAN><SPAN CLASS=S14>VAI</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>R</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">AMS---EE</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S13>F</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S12>EE</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>VM</SPAN><SPAN style="color:#666666">M</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">P</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>LV</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S13>Y</SPAN><SPAN CLASS=S14>GV</SPAN><SPAN CLASS=S7>C</SPAN>
4 PTK7_HUMAN     97.1%  21.2%    <SPAN CLASS=S14>I</SPAN><SPAN CLASS=S18>R</SPAN><SPAN style="color:#666666">EV</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">Q</SPAN><SPAN CLASS=S14>IG</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">A</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">VL</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">TGLS-XLPKGSMNADGVAL</SPAN><SPAN CLASS=S14>VAV</SPAN><SPAN CLASS=S18>KK</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">P</SPAN><SPAN CLASS=S12>D</SPAN><SPAN style="color:#666666">VSD-EVLQ</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S13>F</SPAN><SPAN style="color:#666666">D</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>I</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">D</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>IV</SPAN><SPAN CLASS=S8>Q</SPAN><SPAN CLASS=S14>LLAI</SPAN><SPAN CLASS=S7>C</SPAN>
5 KIN31_CAEEL   100.0%  31.5%    <SPAN CLASS=S14>V</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">LT</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S14>LG</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>GA</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">WK</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">LKILDA-------NHQPVL</SPAN><SPAN CLASS=S14>VAV</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S9>T</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">SMTKEQIK</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>I</SPAN><SPAN style="color:#666666">M</SPAN><SPAN CLASS=S18>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#666666">R</SPAN><SPAN CLASS=S14>LM</SPAN><SPAN style="color:#666666">R</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=S12>D</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=S14>VV</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S13>FF</SPAN><SPAN CLASS=S14>GVA</SPAN>
  consensus/90%                  hp..p.lG.GtFG.V..u.h...................VAlKphp.t........ph.cEh.hM.plpp.plsphhuls
</PRE><p>Notice that the coloured columns correspond to the consensus features (i.e.,
not wildcards or gaps). In each column, the residues that contribute to that
consensus class have been coloured using the prevailing alignment colourmap
(see <a class="reference internal" href="#ref-alignment-colouring"><span class="std std-ref">Alignment colouring</span></a>), which is the default one used in the
other examples in this section.</p>
<p><strong>Mode</strong> <code class="docutils literal notranslate"><span class="pre">group</span></code></p>
<p>The last mode works like the consensus colouring mode, but gives the residues
in a column a uniform colour defined for that consensus class (see
<a class="reference internal" href="#ref-consensus-colouring"><span class="std std-ref">Consensus pattern colouring</span></a>):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">coloring</span> <span class="n">group</span> <span class="o">-</span><span class="n">threshold</span> <span class="mi">90</span>  <span class="o">-</span><span class="n">consensus</span> <span class="n">on</span> <span class="o">-</span><span class="n">con_threshold</span> <span class="mi">90</span>
</pre></div>
</div>
<p>yields:</p>
<PRE>
Colored by: consensus group/90%

                   cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN    100.0% 100.0%    <SPAN CLASS=T14>F</SPAN><SPAN CLASS=T9>K</SPAN><SPAN style="color:#666666">KI</SPAN><SPAN CLASS=T9>K</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=T14>L</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>A</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">T</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">YK</SPAN><SPAN CLASS=T14>G</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=T14>W</SPAN><SPAN style="color:#666666">IPEGEK---------VKIP</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN CLASS=T14>I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>E</SPAN><SPAN CLASS=T14>L</SPAN><SPAN CLASS=T9>R</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=T14>A</SPAN><SPAN style="color:#666666">TSPK-ANK</SPAN><SPAN CLASS=T9>E</SPAN><SPAN CLASS=T14>I</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=T8>D</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=T14>A</SPAN><SPAN style="color:#666666">Y</SPAN><SPAN CLASS=T14>V</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">A</SPAN><SPAN CLASS=T9>S</SPAN><SPAN CLASS=T14>V</SPAN><SPAN CLASS=T9>DN</SPAN><SPAN style="color:#666666">P</SPAN><SPAN CLASS=T9>H</SPAN><SPAN CLASS=T14>VC</SPAN><SPAN CLASS=T9>R</SPAN><SPAN CLASS=T14>LLGIC</SPAN>
2 PR2_DROME      97.1%  35.7%    <SPAN CLASS=T14>I</SPAN><SPAN CLASS=T9>S</SPAN><SPAN style="color:#666666">VN</SPAN><SPAN CLASS=T9>K</SPAN><SPAN style="color:#666666">Q</SPAN><SPAN CLASS=T14>L</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">T</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>E</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">QQ</SPAN><SPAN CLASS=T14>G</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=T14>W</SPAN><SPAN style="color:#666666">SNGNE-----------RIQ</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN CLASS=T14>I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>C</SPAN><SPAN CLASS=T14>L</SPAN><SPAN CLASS=T9>C</SPAN><SPAN style="color:#666666">R</SPAN><SPAN CLASS=T14>E</SPAN><SPAN style="color:#666666">RMQS-NPM</SPAN><SPAN CLASS=T9>E</SPAN><SPAN CLASS=T14>F</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=T8>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=T14>A</SPAN><SPAN style="color:#666666">A</SPAN><SPAN CLASS=T14>I</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">H</SPAN><SPAN CLASS=T9>S</SPAN><SPAN CLASS=T14>I</SPAN><SPAN CLASS=T9>EH</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=T9>N</SPAN><SPAN CLASS=T14>IV</SPAN><SPAN CLASS=T9>R</SPAN><SPAN CLASS=T14>LYGVV</SPAN>
3 ITK_HUMAN      90.0%  32.9%    <SPAN CLASS=T14>L</SPAN><SPAN CLASS=T9>T</SPAN><SPAN style="color:#666666">FV</SPAN><SPAN CLASS=T9>Q</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=T14>I</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>Q</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">HL</SPAN><SPAN CLASS=T14>G</SPAN><SPAN style="color:#666666">Y</SPAN><SPAN CLASS=T14>W</SPAN><SPAN style="color:#666666">LN--------------KDK</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN CLASS=T14>I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>T</SPAN><SPAN CLASS=T14>I</SPAN><SPAN CLASS=T9>R</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=T14>G</SPAN><SPAN style="color:#666666">AMS---EE</SPAN><SPAN CLASS=T9>D</SPAN><SPAN CLASS=T14>F</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=T8>E</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=T14>A</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=T14>V</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">M</SPAN><SPAN CLASS=T9>K</SPAN><SPAN CLASS=T14>L</SPAN><SPAN CLASS=T9>SH</SPAN><SPAN style="color:#666666">P</SPAN><SPAN CLASS=T9>K</SPAN><SPAN CLASS=T14>LV</SPAN><SPAN CLASS=T9>Q</SPAN><SPAN CLASS=T14>LYGVC</SPAN>
4 PTK7_HUMAN     97.1%  21.2%    <SPAN CLASS=T14>I</SPAN><SPAN CLASS=T9>R</SPAN><SPAN style="color:#666666">EV</SPAN><SPAN CLASS=T9>K</SPAN><SPAN style="color:#666666">Q</SPAN><SPAN CLASS=T14>I</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>Q</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">A</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">VL</SPAN><SPAN CLASS=T14>A</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=T14>M</SPAN><SPAN style="color:#666666">TGLS-XLPKGSMNADGVAL</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN CLASS=T14>V</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>K</SPAN><SPAN CLASS=T14>L</SPAN><SPAN CLASS=T9>K</SPAN><SPAN style="color:#666666">P</SPAN><SPAN CLASS=T14>D</SPAN><SPAN style="color:#666666">VSD-EVLQ</SPAN><SPAN CLASS=T9>S</SPAN><SPAN CLASS=T14>F</SPAN><SPAN style="color:#666666">D</SPAN><SPAN CLASS=T8>K</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=T14>I</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=T14>F</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=T9>Q</SPAN><SPAN CLASS=T14>L</SPAN><SPAN CLASS=T9>QH</SPAN><SPAN style="color:#666666">D</SPAN><SPAN CLASS=T9>S</SPAN><SPAN CLASS=T14>IV</SPAN><SPAN CLASS=T9>Q</SPAN><SPAN CLASS=T14>LLAIC</SPAN>
5 KIN31_CAEEL   100.0%  31.5%    <SPAN CLASS=T14>V</SPAN><SPAN CLASS=T9>E</SPAN><SPAN style="color:#666666">LT</SPAN><SPAN CLASS=T9>K</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=T14>L</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>A</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">WK</SPAN><SPAN CLASS=T14>G</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=T14>L</SPAN><SPAN style="color:#666666">LKILDA-------NHQPVL</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN CLASS=T14>V</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>T</SPAN><SPAN CLASS=T14>A</SPAN><SPAN CLASS=T9>K</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=T14>E</SPAN><SPAN style="color:#666666">SMTKEQIK</SPAN><SPAN CLASS=T9>E</SPAN><SPAN CLASS=T14>I</SPAN><SPAN style="color:#666666">M</SPAN><SPAN CLASS=T8>R</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=T14>A</SPAN><SPAN style="color:#666666">R</SPAN><SPAN CLASS=T14>L</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">R</SPAN><SPAN CLASS=T9>N</SPAN><SPAN CLASS=T14>L</SPAN><SPAN CLASS=T9>DH</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=T9>N</SPAN><SPAN CLASS=T14>VV</SPAN><SPAN CLASS=T9>K</SPAN><SPAN CLASS=T14>FFGVA</SPAN>
  consensus/90%                  hp..p.lG.GtFG.V..u.h...................VAlKphp.t........ph.cEh.hM.plpp.plsphhuls
</PRE><p>As in the last example, the coloured columns correspond to the consensus
features (i.e., not wildcards or gaps). In each column, the residues that
contribute to that consensus class have been coloured using a single colour
defined for that consensus class (see <a class="reference internal" href="#ref-consensus-colouring"><span class="std std-ref">Consensus pattern colouring</span></a>), and
conserved residues (at least 90% of a column) are given a solid coloured
background for emphasis.</p>
<p>The choice of consensus classes may be changed using the <code class="docutils literal notranslate"><span class="pre">-groupmap</span> <span class="pre">name</span></code>
option, where <code class="docutils literal notranslate"><span class="pre">name</span></code> is the name of a consensus pattern. These can be listed
using the <code class="docutils literal notranslate"><span class="pre">-listgroups</span></code> option (see <a class="reference internal" href="#ref-changeing-consensus-patterns"><span class="std std-ref">Changing consensus patterns</span></a>).</p>
<p>Note: it is also possible to colour the consensus sequences themselves
independently of the alignment (see <a class="reference internal" href="#ref-consensus-colouring"><span class="std std-ref">Consensus pattern colouring</span></a>).</p>
</section>
<section id="conserved-residue-colouring">
<h3>Conserved residue colouring<a class="headerlink" href="#conserved-residue-colouring" title="Permalink to this headline">¶</a></h3>
<p>The colouring of an alignment under the <code class="docutils literal notranslate"><span class="pre">consensus</span></code> or <code class="docutils literal notranslate"><span class="pre">group</span></code> colouring
modes (see <a class="reference internal" href="#ref-alignment-colouring"><span class="std std-ref">Alignment colouring</span></a>) can be tuned to ignore the
consensus classes with <code class="docutils literal notranslate"><span class="pre">-ignore</span> <span class="pre">class</span></code> for the purposes of colouring:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">coloring</span> <span class="n">group</span> <span class="o">-</span><span class="n">threshold</span> <span class="mi">90</span>  <span class="o">-</span><span class="n">consensus</span> <span class="n">on</span> <span class="o">-</span><span class="n">con_threshold</span> <span class="mi">90</span>  <span class="o">-</span><span class="n">ignore</span> <span class="k">class</span>
</pre></div>
</div>
<p>gives:</p>
<PRE>
Colored by: consensus group/90%

                   cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN    100.0% 100.0%    <SPAN style="color:#666666">FKKIKVL</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">A</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">T</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">YKGLWIPEGEK---------VKIP</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">ELREATSPK-ANKEILD</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">AYV</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">ASVDNPHVCRLLGIC</SPAN>
2 PR2_DROME      97.1%  35.7%    <SPAN style="color:#666666">ISVNKQL</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">T</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">QQGVWSNGNE-----------RIQ</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">CLCRERMQS-NPMEFLK</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">AAI</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">HSIEHENIVRLYGVV</SPAN>
3 ITK_HUMAN      90.0%  32.9%    <SPAN style="color:#666666">LTFVQEI</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">Q</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">HLGYWLN--------------KDK</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">TIREGAMS---EEDFIE</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">AEV</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">MKLSHPKLVQLYGVC</SPAN>
4 PTK7_HUMAN     97.1%  21.2%    <SPAN style="color:#666666">IREVKQI</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">Q</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">A</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">VLAEMTGLS-XLPKGSMNADGVAL</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">KLKPDVSD-EVLQSFDK</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">IKF</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">SQLQHDSIVQLLAIC</SPAN>
5 KIN31_CAEEL   100.0%  31.5%    <SPAN style="color:#666666">VELTKKL</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">A</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">WKGKLLKILDA-------NHQPVL</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">TAKLESMTKEQIKEIMR</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">ARL</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">RNLDHINVVKFFGVA</SPAN>
  consensus/90%                  hp..p.lG.GtFG.V..u.h...................VAlKphp.t........ph.cEh.hM.plpp.plsphhuls
</PRE><p>which highlights the conserved residues (at least 90% of a column) in the
alignment by applying the default consensus group colouring scheme to them.</p>
</section>
<section id="finding-and-colouring-motifs">
<h3>Finding and colouring motifs<a class="headerlink" href="#finding-and-colouring-motifs" title="Permalink to this headline">¶</a></h3>
<p>Occurrences of a string or pattern defined by a regular expression can be
coloured using the <code class="docutils literal notranslate"><span class="pre">-find</span> <span class="pre">'pattern'</span></code> option. This will cause all instances
of the pattern to be highlighted using the user selected colourmap. Patterns
are case-insensitive.</p>
<p><strong>Exact strings</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">html</span> <span class="n">head</span> <span class="o">-</span><span class="n">css</span> <span class="n">on</span> <span class="o">-</span><span class="n">find</span> <span class="n">VAIK</span>
</pre></div>
</div>
<PRE>
Colored by: search pattern 'VAIK'

                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    <SPAN style="color:#666666">FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIP</SPAN><SPAN CLASS=S94>VAIK</SPAN><SPAN style="color:#666666">ELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC</SPAN>
2 PR2_DROME    97.1%  35.7%    <SPAN style="color:#666666">ISVNKQLGTGEFGIVQQGVWSNGNE-----------RIQ</SPAN><SPAN CLASS=S94>VAIK</SPAN><SPAN style="color:#666666">CLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV</SPAN>
3 ITK_HUMAN    90.0%  32.9%    <SPAN style="color:#666666">LTFVQEIGSGQFGLVHLGYWLN--------------KDK</SPAN><SPAN CLASS=S94>VAIK</SPAN><SPAN style="color:#666666">TIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC</SPAN>
4 PTK7_HUMAN   97.1%  21.2%    <SPAN style="color:#666666">IREVKQIGVGQFGAVVLAEMTGLS-XLPKGSMNADGVALVAVKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC</SPAN>
5 KIN31_CAEEL 100.0%  31.5%    <SPAN style="color:#666666">VELTKKLGEGAFGEVWKGKLLKILDA-------NHQPVLVAVKTAKLESMTKEQIKEIMREARLMRNLDHINVVKFFGVA</SPAN>
</PRE><p><strong>Regular expressions</strong></p>
<p>Regular expressions containing any of the shell metacharacters in the set <code class="docutils literal notranslate"><span class="pre">[</span>
<span class="pre">]</span> <span class="pre">{</span> <span class="pre">}</span> <span class="pre">(</span> <span class="pre">)</span> <span class="pre">|</span> <span class="pre">^</span> <span class="pre">$</span> <span class="pre">?</span> <span class="pre">*</span></code> must be enclosed in quotes. This example finds V A
followed by either I or V then K:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">find</span> <span class="s1">&#39;VA[IV]K&#39;</span>
</pre></div>
</div>
<PRE>
Colored by: search pattern 'VA[IV]K'

                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    <SPAN style="color:#666666">FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIP</SPAN><SPAN CLASS=S94>VAIK</SPAN><SPAN style="color:#666666">ELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC</SPAN>
2 PR2_DROME    97.1%  35.7%    <SPAN style="color:#666666">ISVNKQLGTGEFGIVQQGVWSNGNE-----------RIQ</SPAN><SPAN CLASS=S94>VAIK</SPAN><SPAN style="color:#666666">CLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV</SPAN>
3 ITK_HUMAN    90.0%  32.9%    <SPAN style="color:#666666">LTFVQEIGSGQFGLVHLGYWLN--------------KDK</SPAN><SPAN CLASS=S94>VAIK</SPAN><SPAN style="color:#666666">TIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC</SPAN>
4 PTK7_HUMAN   97.1%  21.2%    <SPAN style="color:#666666">IREVKQIGVGQFGAVVLAEMTGLS-XLPKGSMNADGVAL</SPAN><SPAN CLASS=S94>VAVK</SPAN><SPAN style="color:#666666">KLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC</SPAN>
5 KIN31_CAEEL 100.0%  31.5%    <SPAN style="color:#666666">VELTKKLGEGAFGEVWKGKLLKILDA-------NHQPVL</SPAN><SPAN CLASS=S94>VAVK</SPAN><SPAN style="color:#666666">TAKLESMTKEQIKEIMREARLMRNLDHINVVKFFGVA</SPAN>
</PRE><p><strong>Patterns can span gaps in the sequence</strong></p>
<p>The pattern (any 5 residues followed by V A) has intact instances in the last
two rows, but also spans a gap in the first three rows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">find</span> <span class="s1">&#39;.</span><span class="si">{5}</span><span class="s1">VA&#39;</span>
</pre></div>
</div>
<PRE>
Colored by: search pattern '.{5}VA'

                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    <SPAN style="color:#666666">FKKIKVLGSGAFGTVYKGLWIPEGE</SPAN><SPAN CLASS=S94>K</SPAN><SPAN style="color:#666666">---------</SPAN><SPAN CLASS=S94>VKIPVA</SPAN><SPAN style="color:#666666">IKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC</SPAN>
2 PR2_DROME    97.1%  35.7%    <SPAN style="color:#666666">ISVNKQLGTGEFGIVQQGVWSNG</SPAN><SPAN CLASS=S94>NE</SPAN><SPAN style="color:#666666">-----------</SPAN><SPAN CLASS=S94>RIQVA</SPAN><SPAN style="color:#666666">IKCLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV</SPAN>
3 ITK_HUMAN    90.0%  32.9%    <SPAN style="color:#666666">LTFVQEIGSGQFGLVHLGYW</SPAN><SPAN CLASS=S94>LN</SPAN><SPAN style="color:#666666">--------------</SPAN><SPAN CLASS=S94>KDKVA</SPAN><SPAN style="color:#666666">IKTIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC</SPAN>
4 PTK7_HUMAN   97.1%  21.2%    <SPAN style="color:#666666">IREVKQIGVGQFGAVVLAEMTGLS-XLPKGS</SPAN><SPAN CLASS=S94>MNADGVALVA</SPAN><SPAN style="color:#666666">VKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC</SPAN>
5 KIN31_CAEEL 100.0%  31.5%    <SPAN style="color:#666666">VELTKKLGEGAFGEVWKGKLLKILDA-------N</SPAN><SPAN CLASS=S94>HQPVLVA</SPAN><SPAN style="color:#666666">VKTAKLESMTKEQIKEIMREARLMRNLDHINV</SPAN><SPAN CLASS=S94>VKFFGVA</SPAN>
</PRE><p><strong>Patterns find the longest match</strong></p>
<p>Patterns are greedy and try to find the longest possible match. This looks for
a subsequence starting with by V and ending with G:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">find</span> <span class="s1">&#39;V.*G&#39;</span>
</pre></div>
</div>
<PRE>
Colored by: search pattern 'V.*G'

                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    <SPAN style="color:#666666">FKKIK</SPAN><SPAN CLASS=S94>VLGSGAFGTVYKGLWIPEGEK</SPAN><SPAN style="color:#666666">---------</SPAN><SPAN CLASS=S94>VKIPVAIKELREATSPK</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S94>ANKEILDEAYVMASVDNPHVCRLLG</SPAN><SPAN style="color:#666666">IC</SPAN>
2 PR2_DROME    97.1%  35.7%    <SPAN style="color:#666666">IS</SPAN><SPAN CLASS=S94>VNKQLGTGEFGIVQQGVWSNGNE</SPAN><SPAN style="color:#666666">-----------</SPAN><SPAN CLASS=S94>RIQVAIKCLCRERMQS</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S94>NPMEFLKEAAIMHSIEHENIVRLYG</SPAN><SPAN style="color:#666666">VV</SPAN>
3 ITK_HUMAN    90.0%  32.9%    <SPAN style="color:#666666">LTF</SPAN><SPAN CLASS=S94>VQEIGSGQFGLVHLGYWLN</SPAN><SPAN style="color:#666666">--------------</SPAN><SPAN CLASS=S94>KDKVAIKTIREGAMS</SPAN><SPAN style="color:#666666">---</SPAN><SPAN CLASS=S94>EEDFIEEAEVMMKLSHPKLVQLYG</SPAN><SPAN style="color:#666666">VC</SPAN>
4 PTK7_HUMAN   97.1%  21.2%    <SPAN style="color:#666666">IRE</SPAN><SPAN CLASS=S94>VKQIGVGQFGAVVLAEMTGLS</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S94>XLPKGSMNADG</SPAN><SPAN style="color:#666666">VALVAVKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC</SPAN>
5 KIN31_CAEEL 100.0%  31.5%    <SPAN CLASS=S94>VELTKKLGEGAFGEVWKGKLLKILDA</SPAN><SPAN style="color:#666666">-------</SPAN><SPAN CLASS=S94>NHQPVLVAVKTAKLESMTKEQIKEIMREARLMRNLDHINVVKFFG</SPAN><SPAN style="color:#666666">VA</SPAN>
</PRE><p><strong>Alternative patterns</strong></p>
<p>Alternative patterns separated by <code class="docutils literal notranslate"><span class="pre">|</span></code> characters receive the same colouring:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">find</span> <span class="s1">&#39;VAIK|VAVK|[LI]G.G|[DKER]E[AI]..M|[VIL]V[QR]L&#39;</span>
</pre></div>
</div>
<PRE>
Colored by: search pattern 'VAIK|VAVK|[LI]G.G|[DKER]E[AI]..M|[VIL]V[QR]L'

                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    <SPAN style="color:#666666">FKKIKV</SPAN><SPAN CLASS=S94>LGSG</SPAN><SPAN style="color:#666666">AFGTVYKGLWIPEGEK---------VKIP</SPAN><SPAN CLASS=S94>VAIK</SPAN><SPAN style="color:#666666">ELREATSPK-ANKEIL</SPAN><SPAN CLASS=S94>DEAYVM</SPAN><SPAN style="color:#666666">ASVDNPHVCRLLGIC</SPAN>
2 PR2_DROME    97.1%  35.7%    <SPAN style="color:#666666">ISVNKQ</SPAN><SPAN CLASS=S94>LGTG</SPAN><SPAN style="color:#666666">EFGIVQQGVWSNGNE-----------RIQ</SPAN><SPAN CLASS=S94>VAIK</SPAN><SPAN style="color:#666666">CLCRERMQS-NPMEFL</SPAN><SPAN CLASS=S94>KEAAIM</SPAN><SPAN style="color:#666666">HSIEHEN</SPAN><SPAN CLASS=S94>IVRL</SPAN><SPAN style="color:#666666">YGVV</SPAN>
3 ITK_HUMAN    90.0%  32.9%    <SPAN style="color:#666666">LTFVQE</SPAN><SPAN CLASS=S94>IGSG</SPAN><SPAN style="color:#666666">QFGLVHLGYWLN--------------KDK</SPAN><SPAN CLASS=S94>VAIK</SPAN><SPAN style="color:#666666">TIREGAMS---EEDFI</SPAN><SPAN CLASS=S94>EEAEVM</SPAN><SPAN style="color:#666666">MKLSHPK</SPAN><SPAN CLASS=S94>LVQL</SPAN><SPAN style="color:#666666">YGVC</SPAN>
4 PTK7_HUMAN   97.1%  21.2%    <SPAN style="color:#666666">IREVKQ</SPAN><SPAN CLASS=S94>IGVG</SPAN><SPAN style="color:#666666">QFGAVVLAEMTGLS-XLPKGSMNADGVAL</SPAN><SPAN CLASS=S94>VAVK</SPAN><SPAN style="color:#666666">KLKPDVSD-EVLQSFD</SPAN><SPAN CLASS=S94>KEIKFM</SPAN><SPAN style="color:#666666">SQLQHDS</SPAN><SPAN CLASS=S94>IVQL</SPAN><SPAN style="color:#666666">LAIC</SPAN>
5 KIN31_CAEEL 100.0%  31.5%    <SPAN style="color:#666666">VELTKK</SPAN><SPAN CLASS=S94>LGEG</SPAN><SPAN style="color:#666666">AFGEVWKGKLLKILDA-------NHQPVL</SPAN><SPAN CLASS=S94>VAVK</SPAN><SPAN style="color:#666666">TAKLESMTKEQIKEIM</SPAN><SPAN CLASS=S94>REARLM</SPAN><SPAN style="color:#666666">RNLDHINVVKFFGVA</SPAN>
</PRE><p><strong>Alternative patterns with different colours</strong></p>
<p>Alternative patterns separated by <code class="docutils literal notranslate"><span class="pre">:</span></code> characters receive a different colouring:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">find</span> <span class="s1">&#39;VAIK:VAVK:[LI]G.G:[DKER]E[AI]..M:[VIL]V[QR]L&#39;</span>
</pre></div>
</div>
<PRE>
Colored by: search pattern 'VAIK:VAVK:[LI]G.G:[DKER]E[AI]..M:[VIL]V[QR]L'

                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    <SPAN style="color:#666666">FKKIKV</SPAN><SPAN CLASS=S96>LGSG</SPAN><SPAN style="color:#666666">AFGTVYKGLWIPEGEK---------VKIP</SPAN><SPAN CLASS=S94>VAIK</SPAN><SPAN style="color:#666666">ELREATSPK-ANKEIL</SPAN><SPAN CLASS=S97>DEAYVM</SPAN><SPAN style="color:#666666">ASVDNPHVCRLLGIC</SPAN>
2 PR2_DROME    97.1%  35.7%    <SPAN style="color:#666666">ISVNKQ</SPAN><SPAN CLASS=S96>LGTG</SPAN><SPAN style="color:#666666">EFGIVQQGVWSNGNE-----------RIQ</SPAN><SPAN CLASS=S94>VAIK</SPAN><SPAN style="color:#666666">CLCRERMQS-NPMEFL</SPAN><SPAN CLASS=S97>KEAAIM</SPAN><SPAN style="color:#666666">HSIEHEN</SPAN><SPAN CLASS=S98>IVRL</SPAN><SPAN style="color:#666666">YGVV</SPAN>
3 ITK_HUMAN    90.0%  32.9%    <SPAN style="color:#666666">LTFVQE</SPAN><SPAN CLASS=S96>IGSG</SPAN><SPAN style="color:#666666">QFGLVHLGYWLN--------------KDK</SPAN><SPAN CLASS=S94>VAIK</SPAN><SPAN style="color:#666666">TIREGAMS---EEDFI</SPAN><SPAN CLASS=S97>EEAEVM</SPAN><SPAN style="color:#666666">MKLSHPK</SPAN><SPAN CLASS=S98>LVQL</SPAN><SPAN style="color:#666666">YGVC</SPAN>
4 PTK7_HUMAN   97.1%  21.2%    <SPAN style="color:#666666">IREVKQ</SPAN><SPAN CLASS=S96>IGVG</SPAN><SPAN style="color:#666666">QFGAVVLAEMTGLS-XLPKGSMNADGVAL</SPAN><SPAN CLASS=S95>VAVK</SPAN><SPAN style="color:#666666">KLKPDVSD-EVLQSFD</SPAN><SPAN CLASS=S97>KEIKFM</SPAN><SPAN style="color:#666666">SQLQHDS</SPAN><SPAN CLASS=S98>IVQL</SPAN><SPAN style="color:#666666">LAIC</SPAN>
5 KIN31_CAEEL 100.0%  31.5%    <SPAN style="color:#666666">VELTKK</SPAN><SPAN CLASS=S96>LGEG</SPAN><SPAN style="color:#666666">AFGEVWKGKLLKILDA-------NHQPVL</SPAN><SPAN CLASS=S95>VAVK</SPAN><SPAN style="color:#666666">TAKLESMTKEQIKEIM</SPAN><SPAN CLASS=S97>REARLM</SPAN><SPAN style="color:#666666">RNLDHINVVKFFGVA</SPAN>
</PRE><p>Note: if you specify more patterns than the number of colours available
(currently 20) the colours are cycled.</p>
<p><strong>Combining both types of alternative pattern</strong></p>
<p>Patterns containing both <code class="docutils literal notranslate"><span class="pre">|</span></code> and <code class="docutils literal notranslate"><span class="pre">:</span></code> are allowed, with <code class="docutils literal notranslate"><span class="pre">|</span></code> taking
precedence. That is, all neighbouring patterns separated by <code class="docutils literal notranslate"><span class="pre">|</span></code> are treated
as one colour group, and groups separated by <code class="docutils literal notranslate"><span class="pre">:</span></code> are given different
colours:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">find</span> <span class="s1">&#39;VAIK|VAVK|[LI]G.G:[DKER]E[AI]..M|[VIL]V[QR]L&#39;</span>
</pre></div>
</div>
<PRE>
Colored by: search pattern 'VAIK|VAVK|[LI]G.G:[DKER]E[AI]..M|[VIL]V[QR]L'

                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    <SPAN style="color:#666666">FKKIKV</SPAN><SPAN CLASS=S94>LGSG</SPAN><SPAN style="color:#666666">AFGTVYKGLWIPEGEK---------VKIP</SPAN><SPAN CLASS=S94>VAIK</SPAN><SPAN style="color:#666666">ELREATSPK-ANKEIL</SPAN><SPAN CLASS=S95>DEAYVM</SPAN><SPAN style="color:#666666">ASVDNPHVCRLLGIC</SPAN>
2 PR2_DROME    97.1%  35.7%    <SPAN style="color:#666666">ISVNKQ</SPAN><SPAN CLASS=S94>LGTG</SPAN><SPAN style="color:#666666">EFGIVQQGVWSNGNE-----------RIQ</SPAN><SPAN CLASS=S94>VAIK</SPAN><SPAN style="color:#666666">CLCRERMQS-NPMEFL</SPAN><SPAN CLASS=S95>KEAAIM</SPAN><SPAN style="color:#666666">HSIEHEN</SPAN><SPAN CLASS=S95>IVRL</SPAN><SPAN style="color:#666666">YGVV</SPAN>
3 ITK_HUMAN    90.0%  32.9%    <SPAN style="color:#666666">LTFVQE</SPAN><SPAN CLASS=S94>IGSG</SPAN><SPAN style="color:#666666">QFGLVHLGYWLN--------------KDK</SPAN><SPAN CLASS=S94>VAIK</SPAN><SPAN style="color:#666666">TIREGAMS---EEDFI</SPAN><SPAN CLASS=S95>EEAEVM</SPAN><SPAN style="color:#666666">MKLSHPK</SPAN><SPAN CLASS=S95>LVQL</SPAN><SPAN style="color:#666666">YGVC</SPAN>
4 PTK7_HUMAN   97.1%  21.2%    <SPAN style="color:#666666">IREVKQ</SPAN><SPAN CLASS=S94>IGVG</SPAN><SPAN style="color:#666666">QFGAVVLAEMTGLS-XLPKGSMNADGVAL</SPAN><SPAN CLASS=S94>VAVK</SPAN><SPAN style="color:#666666">KLKPDVSD-EVLQSFD</SPAN><SPAN CLASS=S95>KEIKFM</SPAN><SPAN style="color:#666666">SQLQHDS</SPAN><SPAN CLASS=S95>IVQL</SPAN><SPAN style="color:#666666">LAIC</SPAN>
5 KIN31_CAEEL 100.0%  31.5%    <SPAN style="color:#666666">VELTKK</SPAN><SPAN CLASS=S94>LGEG</SPAN><SPAN style="color:#666666">AFGEVWKGKLLKILDA-------NHQPVL</SPAN><SPAN CLASS=S94>VAVK</SPAN><SPAN style="color:#666666">TAKLESMTKEQIKEIM</SPAN><SPAN CLASS=S95>REARLM</SPAN><SPAN style="color:#666666">RNLDHINVVKFFGVA</SPAN>
</PRE></section>
<section id="combining-alignment-and-motif-colouring">
<h3>Combining alignment and motif colouring<a class="headerlink" href="#combining-alignment-and-motif-colouring" title="Permalink to this headline">¶</a></h3>
<p>It is possible to combine the alignment colouring mode with a motif
search. For example, this shows mismatches to the query in a light grey
overlaid with two coloured motifs:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">coloring</span> <span class="n">mismatch</span> <span class="o">-</span><span class="n">colormap</span> <span class="n">gray5</span> <span class="o">-</span><span class="n">find</span> <span class="s1">&#39;G.G.FG.V:VA.K&#39;</span>
</pre></div>
</div>
<PRE>
Colored by: mismatch; search pattern 'G.G.FG.V:VA.K'

                 cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN  100.0% 100.0%    <SPAN style="color:#666666">FKKIKVL</SPAN><SPAN CLASS=S94>GSGAFGTV</SPAN><SPAN style="color:#666666">YKGLWIPEGEK---------VKIP</SPAN><SPAN CLASS=S95>VAIK</SPAN><SPAN style="color:#666666">ELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC</SPAN>
2 PR2_DROME    97.1%  35.7%    <SPAN CLASS=S26>ISVN</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S26>Q</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S94>GTGEFGIV</SPAN><SPAN CLASS=S26>QQ</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S26>V</SPAN><SPAN style="color:#666666">W</SPAN><SPAN CLASS=S26>SNGN</SPAN><SPAN style="color:#666666">E-----------</SPAN><SPAN CLASS=S26>R</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S26>Q</SPAN><SPAN CLASS=S95>VAIK</SPAN><SPAN CLASS=S26>C</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S26>CRERMQS</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S26>NPM</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S26>F</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S26>K</SPAN><SPAN style="color:#666666">EA</SPAN><SPAN CLASS=S26>AI</SPAN><SPAN style="color:#666666">M</SPAN><SPAN CLASS=S26>H</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=S26>IEHENIV</SPAN><SPAN style="color:#666666">RL</SPAN><SPAN CLASS=S26>Y</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S26>VV</SPAN>
3 ITK_HUMAN    90.0%  32.9%    <SPAN CLASS=S26>LTFVQEI</SPAN><SPAN CLASS=S94>GSGQFGLV</SPAN><SPAN CLASS=S26>HL</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S26>Y</SPAN><SPAN style="color:#666666">W</SPAN><SPAN CLASS=S26>LN</SPAN><SPAN style="color:#666666">--------------K</SPAN><SPAN CLASS=S26>DK</SPAN><SPAN CLASS=S95>VAIK</SPAN><SPAN CLASS=S26>TI</SPAN><SPAN style="color:#666666">RE</SPAN><SPAN CLASS=S26>GAMS</SPAN><SPAN style="color:#666666">---</SPAN><SPAN CLASS=S26>EEDFIE</SPAN><SPAN style="color:#666666">EA</SPAN><SPAN CLASS=S26>E</SPAN><SPAN style="color:#666666">VM</SPAN><SPAN CLASS=S26>MKLSH</SPAN><SPAN style="color:#666666">P</SPAN><SPAN CLASS=S26>KLVQ</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S26>Y</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S26>V</SPAN><SPAN style="color:#666666">C</SPAN>
4 PTK7_HUMAN   97.1%  21.2%    <SPAN CLASS=S26>IREV</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S26>QI</SPAN><SPAN CLASS=S94>GVGQFGAV</SPAN><SPAN CLASS=S26>VLAEMTGLS</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S26>XLPKGSMNADGVAL</SPAN><SPAN CLASS=S95>VAVK</SPAN><SPAN CLASS=S26>K</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S26>KPDV</SPAN><SPAN style="color:#666666">S</SPAN><SPAN CLASS=S26>D</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S26>EVLQSFDK</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S26>IKF</SPAN><SPAN style="color:#666666">M</SPAN><SPAN CLASS=S26>SQLQHDSIVQ</SPAN><SPAN style="color:#666666">LL</SPAN><SPAN CLASS=S26>A</SPAN><SPAN style="color:#666666">IC</SPAN>
5 KIN31_CAEEL 100.0%  31.5%    <SPAN CLASS=S26>VELT</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S26>K</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S94>GEGAFGEV</SPAN><SPAN CLASS=S26>W</SPAN><SPAN style="color:#666666">KG</SPAN><SPAN CLASS=S26>KLLKILDA</SPAN><SPAN style="color:#666666">-------</SPAN><SPAN CLASS=S26>NHQPVL</SPAN><SPAN CLASS=S95>VAVK</SPAN><SPAN CLASS=S26>TAKLESMT</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S26>EQI</SPAN><SPAN style="color:#666666">KEI</SPAN><SPAN CLASS=S26>MR</SPAN><SPAN style="color:#666666">EA</SPAN><SPAN CLASS=S26>RL</SPAN><SPAN style="color:#666666">M</SPAN><SPAN CLASS=S26>RNL</SPAN><SPAN style="color:#666666">D</SPAN><SPAN CLASS=S26>HIN</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S26>VKFF</SPAN><SPAN style="color:#666666">G</SPAN><SPAN CLASS=S26>VA</SPAN>
</PRE></section>
<section id="comment-row-colouring">
<span id="ref-comment-colouring"></span><h3>Comment row colouring<a class="headerlink" href="#comment-row-colouring" title="Permalink to this headline">¶</a></h3>
<p>Individual rows can be coloured following a particular scheme by embedding the
name of a colourmap in the sequence identifier of a comment line (see
above). The sequence identifier must begin with a <code class="docutils literal notranslate"><span class="pre">#</span></code> character followed by
the name of a colourmap and a colon separator before the identifier
string. For example, <code class="docutils literal notranslate"><span class="pre">#MAP</span></code> or <code class="docutils literal notranslate"><span class="pre">#MAP:comment</span></code> would cause the MAP
colourmap to be used if it exists. Otherwise, the row will be displayed
uncoloured.</p>
<p>One use of this might be to colour rows of protein secondary structure
assignments. Here is a fragment of an HSSP multiple sequence alignment, which
also shows a DSSP-derived secondary structure assignment under the dummy
identifier <code class="docutils literal notranslate"><span class="pre">#DSSP</span></code>:</p>
<PRE>
                               cov    pid   1 [        .         .         .         .         :         .         .         . 80
1 0:9wga                    100.0% 100.0%     <SPAN style="color:#666666">XRCGEQGSNMECPNNLCCSQYGYCGMGGDYCGKGCQNGACWTSKRCGSQAGGATCPNNHCCSQYGHCGFGAEYCGAGCQG</SPAN>
2 2:uniprot|P02876|agi2_whe  99.4% 100.0%     <SPAN style="color:#666666">-RCGEQGSNMECPNNLCCSQYGYCGMGGDYCGKGCQNGACWTSKRCGSQAGGATCPNNHCCSQYGHCGFGAEYCGAGCQG</SPAN>
3 4:uniprot|P10968|agi1_whe  98.8%  97.6%     <SPAN style="color:#666666">-RCGEQGSNMECPNNLCCSQYGYCGMGGDYCGKGCQNGACWTSKRCGSQAGGATCTNNQCCSQYGYCGFGAEYCGAGCQG</SPAN>
  DSSP                                        <SPAN style="color:#666666">---</SPAN><SPAN CLASS=S9>B</SPAN><SPAN CLASS=S15>GGG</SPAN><SPAN CLASS=S13>TT</SPAN><SPAN CLASS=S9>B</SPAN><SPAN style="color:#666666">--</SPAN><SPAN CLASS=S15>GGG</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S12>EE</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S13>TT</SPAN><SPAN CLASS=S20>S</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S12>EE</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S20>S</SPAN><SPAN CLASS=S18>HHHH</SPAN><SPAN CLASS=S20>S</SPAN><SPAN CLASS=S13>TT</SPAN><SPAN style="color:#666666">--</SPAN><SPAN CLASS=S20>SSS</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S20>SS</SPAN><SPAN style="color:#666666">--</SPAN><SPAN CLASS=S9>B</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S15>GGGG</SPAN><SPAN CLASS=S13>TT</SPAN><SPAN style="color:#666666">---</SPAN><SPAN CLASS=S15>GGG</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S12>EE</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S13>TT</SPAN><SPAN CLASS=S20>S</SPAN><SPAN CLASS=S9>B</SPAN><SPAN CLASS=S12>EE</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S20>S</SPAN><SPAN CLASS=S18>HHHH</SPAN><SPAN CLASS=S20>S</SPAN><SPAN CLASS=S13>TT</SPAN><SPAN style="color:#666666">--</SPAN><SPAN CLASS=S9>B</SPAN>
</PRE><p>Note 1: MView strips the <code class="docutils literal notranslate"><span class="pre">#</span></code> character from the row identifier in the
display, but leaves the name of the colourmap. The intention is that the user
can define various specialized colourmaps with informative names, as here with
DSSP.</p>
<p>Note 2: Any number of these rows is allowed, using the same or different
colourmaps, and they can be placed anywhere in the alignment.</p>
</section>
<section id="consensus-pattern-colouring">
<span id="ref-consensus-colouring"></span><h3>Consensus pattern colouring<a class="headerlink" href="#consensus-pattern-colouring" title="Permalink to this headline">¶</a></h3>
<p>The consensus lines may be coloured independently of the alignment. The
<code class="docutils literal notranslate"><span class="pre">--con_coloring</span></code> option takes two modes: <code class="docutils literal notranslate"><span class="pre">any</span></code>, <code class="docutils literal notranslate"><span class="pre">identity</span></code>.</p>
<p><strong>Mode</strong> <code class="docutils literal notranslate"><span class="pre">any</span></code></p>
<p>The simplest is to colour every consensus symbol according to the currently
selected consensus colourmap:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">consensus</span> <span class="n">on</span> <span class="o">-</span><span class="n">con_coloring</span> <span class="nb">any</span>
</pre></div>
</div>
<p>gives:</p>
<PRE>
                    cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN     100.0% 100.0%    FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIPVAIKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC
2 PR2_DROME       97.1%  35.7%    ISVNKQLGTGEFGIVQQGVWSNGNE-----------RIQVAIKCLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV
3 ITK_HUMAN       90.0%  32.9%    LTFVQEIGSGQFGLVHLGYWLN--------------KDKVAIKTIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC
4 PTK7_HUMAN      97.1%  21.2%    IREVKQIGVGQFGAVVLAEMTGLS-XLPKGSMNADGVALVAVKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC
5 KIN31_CAEEL    100.0%  31.5%    VELTKKLGEGAFGEVWKGKLLKILDA-------NHQPVLVAVKTAKLESMTKEQIKEIMREARLMRNLDHINVVKFFGVA
  consensus/100%                  <SPAN CLASS=T14>h</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>t</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T14>u</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">...................</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>t</SPAN><SPAN style="color:#666666">........</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T8>c</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>pp</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>ls</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>hhuls</SPAN>
  consensus/90%                   <SPAN CLASS=T14>h</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>t</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T14>u</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">...................</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>t</SPAN><SPAN style="color:#666666">........</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T8>c</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>pp</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>ls</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>hhuls</SPAN>
  consensus/80%                   <SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>hs</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>s</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>t</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>hhhs</SPAN><SPAN style="color:#666666">..............</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T18>+</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>ts</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>s</SPAN><SPAN style="color:#666666">....</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T12>-</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN CLASS=T8>c</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=T14>tl</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=T14>t</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>ls</SPAN>
  consensus/70%                   <SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>hs</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>s</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>t</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>hhhs</SPAN><SPAN style="color:#666666">..............</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T18>+</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>ts</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>s</SPAN><SPAN style="color:#666666">....</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T12>-</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN CLASS=T8>c</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=T14>tl</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=T14>t</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>ls</SPAN>
</PRE><p>Alternative colourmaps can be set using the <code class="docutils literal notranslate"><span class="pre">-con_colormap</span></code> option. In
particular, a predefined uniform colour can be chosen:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">consensus</span> <span class="n">on</span> <span class="o">-</span><span class="n">con_coloring</span> <span class="nb">any</span> <span class="o">-</span><span class="n">con_colormap</span> <span class="n">brown</span>
</pre></div>
</div>
<p>gives:</p>
<PRE>
                    cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN     100.0% 100.0%    FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIPVAIKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC
2 PR2_DROME       97.1%  35.7%    ISVNKQLGTGEFGIVQQGVWSNGNE-----------RIQVAIKCLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV
3 ITK_HUMAN       90.0%  32.9%    LTFVQEIGSGQFGLVHLGYWLN--------------KDKVAIKTIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC
4 PTK7_HUMAN      97.1%  21.2%    IREVKQIGVGQFGAVVLAEMTGLS-XLPKGSMNADGVALVAVKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC
5 KIN31_CAEEL    100.0%  31.5%    VELTKKLGEGAFGEVWKGKLLKILDA-------NHQPVLVAVKTAKLESMTKEQIKEIMREARLMRNLDHINVVKFFGVA
  consensus/100%                  <SPAN CLASS=T17>hp</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T17>p</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>lG</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>GtFG</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>V</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T17>u</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>h</SPAN><SPAN style="color:#666666">...................</SPAN><SPAN CLASS=T17>VAlKphp</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>t</SPAN><SPAN style="color:#666666">........</SPAN><SPAN CLASS=T17>ph</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>cEh</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>hM</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>plpp</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>plsphhuls</SPAN>
  consensus/90%                   <SPAN CLASS=T17>hp</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T17>p</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>lG</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>GtFG</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>V</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T17>u</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>h</SPAN><SPAN style="color:#666666">...................</SPAN><SPAN CLASS=T17>VAlKphp</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>t</SPAN><SPAN style="color:#666666">........</SPAN><SPAN CLASS=T17>ph</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>cEh</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>hM</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>plpp</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>plsphhuls</SPAN>
  consensus/80%                   <SPAN CLASS=T17>lphsKplGsGtFGhVhhGhhhs</SPAN><SPAN style="color:#666666">..............</SPAN><SPAN CLASS=T17>hh</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>VAlKpl+</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>ts</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>s</SPAN><SPAN style="color:#666666">....</SPAN><SPAN CLASS=T17>p-hhcEAtlMtplpH</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>plVpLhGls</SPAN>
  consensus/70%                   <SPAN CLASS=T17>lphsKplGsGtFGhVhhGhhhs</SPAN><SPAN style="color:#666666">..............</SPAN><SPAN CLASS=T17>hh</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>VAlKpl+</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>ts</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>s</SPAN><SPAN style="color:#666666">....</SPAN><SPAN CLASS=T17>p-hhcEAtlMtplpH</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T17>plVpLhGls</SPAN>
</PRE><p><strong>Mode</strong> <code class="docutils literal notranslate"><span class="pre">identity</span></code></p>
<p>Instead of colouring all the consensus symbols, we can just colour those that
are identical to the reference sequence. Here we colour the consensus
identities with respect to the first alignment row:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">consensus</span> <span class="n">on</span> <span class="o">-</span><span class="n">con_coloring</span> <span class="n">identity</span>
</pre></div>
</div>
<p>gives:</p>
<PRE>
                    cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN     100.0% 100.0%    FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIPVAIKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC
2 PR2_DROME       97.1%  35.7%    ISVNKQLGTGEFGIVQQGVWSNGNE-----------RIQVAIKCLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV
3 ITK_HUMAN       90.0%  32.9%    LTFVQEIGSGQFGLVHLGYWLN--------------KDKVAIKTIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC
4 PTK7_HUMAN      97.1%  21.2%    IREVKQIGVGQFGAVVLAEMTGLS-XLPKGSMNADGVALVAVKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC
5 KIN31_CAEEL    100.0%  31.5%    VELTKKLGEGAFGEVWKGKLLKILDA-------NHQPVLVAVKTAKLESMTKEQIKEIMREARLMRNLDHINVVKFFGVA
  consensus/100%                  <SPAN style="color:#000000">hp</SPAN><SPAN style="color:#666666">..</SPAN><SPAN style="color:#000000">p</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">l</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#000000">t</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">..</SPAN><SPAN style="color:#000000">u</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">h</SPAN><SPAN style="color:#666666">...................</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#000000">l</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#000000">php</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">t</SPAN><SPAN style="color:#666666">........</SPAN><SPAN style="color:#000000">ph</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">c</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#000000">h</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">h</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">plpp</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">plsphhuls</SPAN>
  consensus/90%                   <SPAN style="color:#000000">hp</SPAN><SPAN style="color:#666666">..</SPAN><SPAN style="color:#000000">p</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">l</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#000000">t</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">..</SPAN><SPAN style="color:#000000">u</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">h</SPAN><SPAN style="color:#666666">...................</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#000000">l</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#000000">php</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">t</SPAN><SPAN style="color:#666666">........</SPAN><SPAN style="color:#000000">ph</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">c</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#000000">h</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">h</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">plpp</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">plsphhuls</SPAN>
  consensus/80%                   <SPAN style="color:#000000">lphs</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#000000">pl</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#000000">s</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#000000">t</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#000000">h</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#000000">hh</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#000000">hhhs</SPAN><SPAN style="color:#666666">..............</SPAN><SPAN style="color:#000000">hh</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#000000">l</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#000000">pl+</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">ts</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">s</SPAN><SPAN style="color:#666666">....</SPAN><SPAN style="color:#000000">p-hhc</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#000000">tl</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#000000">tplp</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">pl</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#000000">p</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#000000">h</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#000000">ls</SPAN>
  consensus/70%                   <SPAN style="color:#000000">lphs</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#000000">pl</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#000000">s</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#000000">t</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#000000">h</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#000000">hh</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#000000">hhhs</SPAN><SPAN style="color:#666666">..............</SPAN><SPAN style="color:#000000">hh</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#000000">l</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#000000">pl+</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">ts</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">s</SPAN><SPAN style="color:#666666">....</SPAN><SPAN style="color:#000000">p-hhc</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#000000">tl</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#000000">tplp</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">.</SPAN><SPAN style="color:#000000">pl</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#000000">p</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#000000">h</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#000000">ls</SPAN>
</PRE><p>These modes work with the <code class="docutils literal notranslate"><span class="pre">-con_gaps</span></code> and <code class="docutils literal notranslate"><span class="pre">-con_ignore</span></code> options in various
combinations to tune the consensus symbols displayed (see
<a class="reference internal" href="#ref-conserved-symbols-or-conserved-classes"><span class="std std-ref">Showing conserved residues or conserved classes</span></a>).</p>
<p><strong>Make consensus span low coverage regions</strong></p>
<p>The strict consensus calculation can be relaxed to ignore gaps, so that the
consensus is also computed across lower coverage regions:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">consensus</span> <span class="n">on</span> <span class="o">-</span><span class="n">con_coloring</span> <span class="nb">any</span> <span class="o">-</span><span class="n">con_gaps</span> <span class="n">off</span>
</pre></div>
</div>
<p>to produce:</p>
<PRE>
                    cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN     100.0% 100.0%    FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIPVAIKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC
2 PR2_DROME       97.1%  35.7%    ISVNKQLGTGEFGIVQQGVWSNGNE-----------RIQVAIKCLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV
3 ITK_HUMAN       90.0%  32.9%    LTFVQEIGSGQFGLVHLGYWLN--------------KDKVAIKTIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC
4 PTK7_HUMAN      97.1%  21.2%    IREVKQIGVGQFGAVVLAEMTGLS-XLPKGSMNADGVALVAVKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC
5 KIN31_CAEEL    100.0%  31.5%    VELTKKLGEGAFGEVWKGKLLKILDA-------NHQPVLVAVKTAKLESMTKEQIKEIMREARLMRNLDHINVVKFFGVA
  consensus/100%                  <SPAN CLASS=T14>h</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>t</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T14>u</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">....</SPAN><SPAN CLASS=T12>-</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>LP</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=T14>s</SPAN><SPAN CLASS=T8>c</SPAN><SPAN style="color:#666666">....</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>t</SPAN><SPAN style="color:#666666">...</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">...</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T8>c</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>pp</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>ls</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>hhuls</SPAN>
  consensus/90%                   <SPAN CLASS=T14>h</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>t</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T14>u</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">....</SPAN><SPAN CLASS=T12>-</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>LP</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=T14>s</SPAN><SPAN CLASS=T8>c</SPAN><SPAN style="color:#666666">....</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>t</SPAN><SPAN style="color:#666666">...</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">...</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T8>c</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>pp</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>ls</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>hhuls</SPAN>
  consensus/80%                   <SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>hs</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>s</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>t</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>hhhs</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T12>-</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>LP</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=T14>s</SPAN><SPAN CLASS=T8>c</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T18>+</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>ts</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>s</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T12>-</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN CLASS=T8>c</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=T14>tl</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=T14>t</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>ls</SPAN>
  consensus/70%                   <SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>hs</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>s</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>t</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>hhhshs</SPAN><SPAN CLASS=T12>-</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>LP</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=S9>S</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=S8>N</SPAN><SPAN CLASS=T14>s</SPAN><SPAN CLASS=T8>c</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S18>K</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T18>+</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>ts</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>s</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=T14>s</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T12>-</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN CLASS=T8>c</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN CLASS=T14>tl</SPAN><SPAN CLASS=S14>M</SPAN><SPAN CLASS=T14>t</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=S14>V</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=S14>L</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=S14>G</SPAN><SPAN CLASS=T14>ls</SPAN>
</PRE><p><strong>Make consensus span low coverage regions, switch off conserved residues</strong></p>
<p>The strict consensus calculation can be relaxed to ignore gaps, so that the
consensus is also computed across lower coverage regions. In addition,
conserved residue names can be replaced with their consensus classes:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">consensus</span> <span class="n">on</span> <span class="o">-</span><span class="n">con_coloring</span> <span class="nb">any</span> <span class="o">-</span><span class="n">con_gaps</span> <span class="n">off</span> <span class="o">-</span><span class="n">con_ignore</span> <span class="n">singleton</span>
</pre></div>
</div>
<p>to produce:</p>
<PRE>
                    cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN     100.0% 100.0%    FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIPVAIKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC
2 PR2_DROME       97.1%  35.7%    ISVNKQLGTGEFGIVQQGVWSNGNE-----------RIQVAIKCLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV
3 ITK_HUMAN       90.0%  32.9%    LTFVQEIGSGQFGLVHLGYWLN--------------KDKVAIKTIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC
4 PTK7_HUMAN      97.1%  21.2%    IREVKQIGVGQFGAVVLAEMTGLS-XLPKGSMNADGVALVAVKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC
5 KIN31_CAEEL    100.0%  31.5%    VELTKKLGEGAFGEVWKGKLLKILDA-------NHQPVLVAVKTAKLESMTKEQIKEIMREARLMRNLDHINVVKFFGVA
  consensus/100%                  <SPAN CLASS=T14>h</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>lu</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>ut</SPAN><SPAN CLASS=T13>a</SPAN><SPAN CLASS=T14>u</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>l</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T14>u</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">....</SPAN><SPAN CLASS=T12>-</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>ls</SPAN><SPAN CLASS=T18>+</SPAN><SPAN CLASS=T14>u</SPAN><SPAN CLASS=T9>o</SPAN><SPAN CLASS=T14>hss</SPAN><SPAN CLASS=T8>c</SPAN><SPAN style="color:#666666">....</SPAN><SPAN CLASS=T14>lul</SPAN><SPAN CLASS=T18>+</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>t</SPAN><SPAN style="color:#666666">...</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T12>-</SPAN><SPAN style="color:#666666">...</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T8>c</SPAN><SPAN CLASS=T12>-</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>pp</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>ls</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>hhuls</SPAN>
  consensus/90%                   <SPAN CLASS=T14>h</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>lu</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>ut</SPAN><SPAN CLASS=T13>a</SPAN><SPAN CLASS=T14>u</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>l</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T14>u</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">....</SPAN><SPAN CLASS=T12>-</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>ls</SPAN><SPAN CLASS=T18>+</SPAN><SPAN CLASS=T14>u</SPAN><SPAN CLASS=T9>o</SPAN><SPAN CLASS=T14>hss</SPAN><SPAN CLASS=T8>c</SPAN><SPAN style="color:#666666">....</SPAN><SPAN CLASS=T14>lul</SPAN><SPAN CLASS=T18>+</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>h</SPAN><SPAN CLASS=T9>p</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>t</SPAN><SPAN style="color:#666666">...</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T12>-</SPAN><SPAN style="color:#666666">...</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T8>c</SPAN><SPAN CLASS=T12>-</SPAN><SPAN CLASS=T14>h</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>pp</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>ls</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>hhuls</SPAN>
  consensus/80%                   <SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>hs</SPAN><SPAN CLASS=T18>+</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>lusut</SPAN><SPAN CLASS=T13>a</SPAN><SPAN CLASS=T14>uhlhhuhhhs</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T12>-</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>ls</SPAN><SPAN CLASS=T18>+</SPAN><SPAN CLASS=T14>u</SPAN><SPAN CLASS=T9>o</SPAN><SPAN CLASS=T14>hss</SPAN><SPAN CLASS=T8>c</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>lul</SPAN><SPAN CLASS=T18>+</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T18>+</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>ts</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>s</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T12>-</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T12>-</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN CLASS=T8>c</SPAN><SPAN CLASS=T12>-</SPAN><SPAN CLASS=T14>utlht</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T18>+</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>ll</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>lhuls</SPAN>
  consensus/70%                   <SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>hs</SPAN><SPAN CLASS=T18>+</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>lusut</SPAN><SPAN CLASS=T13>a</SPAN><SPAN CLASS=T14>uhlhhuhhhshs</SPAN><SPAN CLASS=T12>-</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>ls</SPAN><SPAN CLASS=T18>+</SPAN><SPAN CLASS=T14>u</SPAN><SPAN CLASS=T9>o</SPAN><SPAN CLASS=T14>hss</SPAN><SPAN CLASS=T8>c</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>lul</SPAN><SPAN CLASS=T18>+</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T18>+</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>ts</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T14>s</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T12>-</SPAN><SPAN CLASS=T14>s</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T12>-</SPAN><SPAN CLASS=T14>hh</SPAN><SPAN CLASS=T8>c</SPAN><SPAN CLASS=T12>-</SPAN><SPAN CLASS=T14>utlht</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>l</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T18>+</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>ll</SPAN><SPAN CLASS=T9>p</SPAN><SPAN CLASS=T14>lhuls</SPAN>
</PRE><p><strong>Switch off consensus classes, show only conserved residues</strong></p>
<p>The consensus classes can be switched off, to leave only conserved residues:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">consensus</span> <span class="n">on</span> <span class="o">-</span><span class="n">con_coloring</span> <span class="nb">any</span> <span class="o">-</span><span class="n">con_ignore</span> <span class="k">class</span>
</pre></div>
</div>
<p>to produce:</p>
<PRE>
                    cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 EGFR_HUMAN     100.0% 100.0%    FKKIKVLGSGAFGTVYKGLWIPEGEK---------VKIPVAIKELREATSPK-ANKEILDEAYVMASVDNPHVCRLLGIC
2 PR2_DROME       97.1%  35.7%    ISVNKQLGTGEFGIVQQGVWSNGNE-----------RIQVAIKCLCRERMQS-NPMEFLKEAAIMHSIEHENIVRLYGVV
3 ITK_HUMAN       90.0%  32.9%    LTFVQEIGSGQFGLVHLGYWLN--------------KDKVAIKTIREGAMS---EEDFIEEAEVMMKLSHPKLVQLYGVC
4 PTK7_HUMAN      97.1%  21.2%    IREVKQIGVGQFGAVVLAEMTGLS-XLPKGSMNADGVALVAVKKLKPDVSD-EVLQSFDKEIKFMSQLQHDSIVQLLAIC
5 KIN31_CAEEL    100.0%  31.5%    VELTKKLGEGAFGEVWKGKLLKILDA-------NHQPVLVAVKTAKLESMTKEQIKEIMREARLMRNLDHINVVKFFGVA
  consensus/100%                  <SPAN style="color:#666666">.......</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">........................</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">.................</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">...</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">...............</SPAN>
  consensus/90%                   <SPAN style="color:#666666">.......</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">........................</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">.................</SPAN><SPAN CLASS=S12>E</SPAN><SPAN style="color:#666666">...</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">...............</SPAN>
  consensus/80%                   <SPAN style="color:#666666">....</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.....................</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">.................</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">....</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">...</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">..</SPAN>
  consensus/70%                   <SPAN style="color:#666666">....</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S13>F</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">.....................</SPAN><SPAN CLASS=S14>VA</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S18>K</SPAN><SPAN style="color:#666666">.................</SPAN><SPAN CLASS=S12>E</SPAN><SPAN CLASS=S14>A</SPAN><SPAN style="color:#666666">..</SPAN><SPAN CLASS=S14>M</SPAN><SPAN style="color:#666666">....</SPAN><SPAN CLASS=S13>H</SPAN><SPAN style="color:#666666">...</SPAN><SPAN CLASS=S14>V</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>L</SPAN><SPAN style="color:#666666">.</SPAN><SPAN CLASS=S14>G</SPAN><SPAN style="color:#666666">..</SPAN>
</PRE></section>
</section>
<section id="colours">
<span id="ref-colourmaps"></span><h2>Colours<a class="headerlink" href="#colours" title="Permalink to this headline">¶</a></h2>
<section id="colourmaps">
<h3>Colourmaps<a class="headerlink" href="#colourmaps" title="Permalink to this headline">¶</a></h3>
<p>There are default colourmaps for protein and nucleotide (either DNA or RNA)
alignments and consensus lines. MView starts up with the default protein
colourmap selected, as if you had specified a molecule type with <code class="docutils literal notranslate"><span class="pre">-moltype</span>
<span class="pre">aa</span></code> (for “amino acid”).</p>
<p><strong>Alignments</strong></p>
<p>Colourmaps have names, e.g., the default protein alignment colourmap is called
<code class="docutils literal notranslate"><span class="pre">P1</span></code> and the default nucleotide colourmap is <code class="docutils literal notranslate"><span class="pre">D1</span></code>. Alternative alignment
colouring colourmaps are explicitly selected using the <code class="docutils literal notranslate"><span class="pre">-colormap</span></code>
option. For example, another built-in colouring scheme can be specified with
<code class="docutils literal notranslate"><span class="pre">-colormap</span> <span class="pre">CLUSTAL</span></code>.</p>
<p>Here is the default colormap for proteins:</p>
<PRE>
<SPAN style="color:#000000">[P1]</SPAN>
<SPAN style="color:#aa6666">#Protein: highlight amino acid physicochemical properties
#symbols =>  color                #comment</SPAN>
<SPAN style="color:#666666">.      </SPAN>  ->  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#wildcard/mismatch</SPAN>
<SPAN style="color:#33cc00">Aa     </SPAN>  =>  <SPAN style="color:#33cc00">bright-green        </SPAN> <SPAN style="color:#aa6666">#hydrophobic</SPAN>
<SPAN style="color:#666666">Bb     </SPAN>  =>  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#D or N</SPAN>
<SPAN style="color:#ffff00">Cc     </SPAN>  =>  <SPAN style="color:#ffff00">yellow              </SPAN> <SPAN style="color:#aa6666">#cysteine</SPAN>
<SPAN style="color:#0033ff">Dd     </SPAN>  =>  <SPAN style="color:#0033ff">bright-blue         </SPAN> <SPAN style="color:#aa6666">#negative charge</SPAN>
<SPAN style="color:#0033ff">Ee     </SPAN>  =>  <SPAN style="color:#0033ff">bright-blue         </SPAN> <SPAN style="color:#aa6666">#negative charge</SPAN>
<SPAN style="color:#009900">Ff     </SPAN>  =>  <SPAN style="color:#009900">dark-green          </SPAN> <SPAN style="color:#aa6666">#large hydrophobic</SPAN>
<SPAN style="color:#33cc00">Gg     </SPAN>  =>  <SPAN style="color:#33cc00">bright-green        </SPAN> <SPAN style="color:#aa6666">#hydrophobic</SPAN>
<SPAN style="color:#009900">Hh     </SPAN>  =>  <SPAN style="color:#009900">dark-green          </SPAN> <SPAN style="color:#aa6666">#large hydrophobic</SPAN>
<SPAN style="color:#33cc00">Ii     </SPAN>  =>  <SPAN style="color:#33cc00">bright-green        </SPAN> <SPAN style="color:#aa6666">#hydrophobic</SPAN>
<SPAN style="color:#cc0000">Kk     </SPAN>  =>  <SPAN style="color:#cc0000">bright-red          </SPAN> <SPAN style="color:#aa6666">#positive charge</SPAN>
<SPAN style="color:#33cc00">Ll     </SPAN>  =>  <SPAN style="color:#33cc00">bright-green        </SPAN> <SPAN style="color:#aa6666">#hydrophobic</SPAN>
<SPAN style="color:#33cc00">Mm     </SPAN>  =>  <SPAN style="color:#33cc00">bright-green        </SPAN> <SPAN style="color:#aa6666">#hydrophobic</SPAN>
<SPAN style="color:#6600cc">Nn     </SPAN>  =>  <SPAN style="color:#6600cc">purple              </SPAN> <SPAN style="color:#aa6666">#polar</SPAN>
<SPAN style="color:#33cc00">Pp     </SPAN>  =>  <SPAN style="color:#33cc00">bright-green        </SPAN> <SPAN style="color:#aa6666">#hydrophobic</SPAN>
<SPAN style="color:#6600cc">Qq     </SPAN>  =>  <SPAN style="color:#6600cc">purple              </SPAN> <SPAN style="color:#aa6666">#polar</SPAN>
<SPAN style="color:#cc0000">Rr     </SPAN>  =>  <SPAN style="color:#cc0000">bright-red          </SPAN> <SPAN style="color:#aa6666">#positive charge</SPAN>
<SPAN style="color:#0099ff">Ss     </SPAN>  =>  <SPAN style="color:#0099ff">dull-blue           </SPAN> <SPAN style="color:#aa6666">#small alcohol</SPAN>
<SPAN style="color:#0099ff">Tt     </SPAN>  =>  <SPAN style="color:#0099ff">dull-blue           </SPAN> <SPAN style="color:#aa6666">#small alcohol</SPAN>
<SPAN style="color:#33cc00">Vv     </SPAN>  =>  <SPAN style="color:#33cc00">bright-green        </SPAN> <SPAN style="color:#aa6666">#hydrophobic</SPAN>
<SPAN style="color:#009900">Ww     </SPAN>  =>  <SPAN style="color:#009900">dark-green          </SPAN> <SPAN style="color:#aa6666">#large hydrophobic</SPAN>
<SPAN style="color:#009900">Yy     </SPAN>  =>  <SPAN style="color:#009900">dark-green          </SPAN> <SPAN style="color:#aa6666">#large hydrophobic</SPAN>
<SPAN style="color:#666666">Zz     </SPAN>  =>  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#E or Q</SPAN>
<SPAN style="color:#666666">Xx     </SPAN>  ->  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#unknown</SPAN>
<SPAN style="color:#999999">?      </SPAN>  ->  <SPAN style="color:#999999">light-gray          </SPAN> <SPAN style="color:#aa6666">#unknown</SPAN>
<SPAN style="color:#000000">*      </SPAN>  ->  <SPAN style="color:#000000">black               </SPAN> <SPAN style="color:#aa6666">#stop</SPAN>
</PRE><p>The default alignment colormap for nucleotide sequences can be selected using
the <code class="docutils literal notranslate"><span class="pre">-moltype</span> <span class="pre">na</span></code> (or <code class="docutils literal notranslate"><span class="pre">dna</span></code> or <code class="docutils literal notranslate"><span class="pre">rna</span></code>) option:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">moltype</span> <span class="n">na</span>
</pre></div>
</div>
<p>or by specifying:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">colormap</span> <span class="n">D1</span>
</pre></div>
</div>
<p>and is defined as:</p>
<PRE>
<SPAN style="color:#000000">[D1]</SPAN>
<SPAN style="color:#aa6666">#DNA: highlight nucleotide types
#symbols =>  color                #comment</SPAN>
<SPAN style="color:#666666">.      </SPAN>  ->  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#wildcard/mismatch</SPAN>
<SPAN style="color:#0033ff">Aa     </SPAN>  =>  <SPAN style="color:#0033ff">bright-blue         </SPAN> <SPAN style="color:#aa6666">#adenosine</SPAN>
<SPAN style="color:#0099ff">Cc     </SPAN>  =>  <SPAN style="color:#0099ff">dull-blue           </SPAN> <SPAN style="color:#aa6666">#cytosine</SPAN>
<SPAN style="color:#0033ff">Gg     </SPAN>  =>  <SPAN style="color:#0033ff">bright-blue         </SPAN> <SPAN style="color:#aa6666">#guanine</SPAN>
<SPAN style="color:#0099ff">Tt     </SPAN>  =>  <SPAN style="color:#0099ff">dull-blue           </SPAN> <SPAN style="color:#aa6666">#thymine</SPAN>
<SPAN style="color:#0099ff">Uu     </SPAN>  =>  <SPAN style="color:#0099ff">dull-blue           </SPAN> <SPAN style="color:#aa6666">#uracil</SPAN>
<SPAN style="color:#666666">Mm     </SPAN>  =>  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#amino:      A or C</SPAN>
<SPAN style="color:#666666">Rr     </SPAN>  =>  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#purine:     A or G</SPAN>
<SPAN style="color:#666666">Ww     </SPAN>  =>  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#weak:       A or T</SPAN>
<SPAN style="color:#666666">Ss     </SPAN>  =>  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#strong:     C or G</SPAN>
<SPAN style="color:#666666">Yy     </SPAN>  =>  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#pyrimidine: C or T</SPAN>
<SPAN style="color:#666666">Kk     </SPAN>  =>  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#keto:       G or T</SPAN>
<SPAN style="color:#666666">Vv     </SPAN>  =>  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#not T: A or C or G</SPAN>
<SPAN style="color:#666666">Hh     </SPAN>  =>  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#not G: A or C or T</SPAN>
<SPAN style="color:#666666">Dd     </SPAN>  =>  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#not C: A or G or T</SPAN>
<SPAN style="color:#666666">Bb     </SPAN>  =>  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#not A: C or G or T</SPAN>
<SPAN style="color:#666666">Nn     </SPAN>  =>  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#any: A or C or G or T</SPAN>
<SPAN style="color:#666666">Xx     </SPAN>  ->  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#any</SPAN>
<SPAN style="color:#999999">?      </SPAN>  ->  <SPAN style="color:#999999">light-gray          </SPAN> <SPAN style="color:#aa6666">#unknown</SPAN>
</PRE><p><strong>Consensus lines</strong></p>
<p>In addition, the consensus lines optionally displayed below an alignment can
be coloured, and they have their own consensus colourmaps; the default for
proteins is <code class="docutils literal notranslate"><span class="pre">PC1</span></code> and for nucleotides it is <code class="docutils literal notranslate"><span class="pre">DC1</span></code>. Alternative consensus
colouring colourmaps are explicitly selected using the <code class="docutils literal notranslate"><span class="pre">-con_colormap</span></code> option.</p>
<p>Here is the default consensus colormap for proteins:</p>
<PRE>
<SPAN style="color:#000000">[PC1]</SPAN>
<SPAN style="color:#aa6666">#Protein consensus: highlight equivalence class
#symbols =>  color                #comment</SPAN>
<SPAN style="color:#666666">.      </SPAN>  ->  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#unconserved</SPAN>
<SPAN style="color:#cc0000">+      </SPAN>  ->  <SPAN style="color:#cc0000">bright-red          </SPAN> <SPAN style="color:#aa6666">#positive charge</SPAN>
<SPAN style="color:#0033ff">-      </SPAN>  ->  <SPAN style="color:#0033ff">bright-blue         </SPAN> <SPAN style="color:#aa6666">#negative charge</SPAN>
<SPAN style="color:#009900">a      </SPAN>  ->  <SPAN style="color:#009900">dark-green          </SPAN> <SPAN style="color:#aa6666">#aromatic</SPAN>
<SPAN style="color:#6600cc">c      </SPAN>  ->  <SPAN style="color:#6600cc">purple              </SPAN> <SPAN style="color:#aa6666">#charged</SPAN>
<SPAN style="color:#33cc00">h      </SPAN>  ->  <SPAN style="color:#33cc00">bright-green        </SPAN> <SPAN style="color:#aa6666">#hydrophobic</SPAN>
<SPAN style="color:#33cc00">l      </SPAN>  ->  <SPAN style="color:#33cc00">bright-green        </SPAN> <SPAN style="color:#aa6666">#aliphatic</SPAN>
<SPAN style="color:#0099ff">o      </SPAN>  ->  <SPAN style="color:#0099ff">dull-blue           </SPAN> <SPAN style="color:#aa6666">#alcohol</SPAN>
<SPAN style="color:#0099ff">p      </SPAN>  ->  <SPAN style="color:#0099ff">dull-blue           </SPAN> <SPAN style="color:#aa6666">#polar</SPAN>
<SPAN style="color:#33cc00">s      </SPAN>  ->  <SPAN style="color:#33cc00">bright-green        </SPAN> <SPAN style="color:#aa6666">#small</SPAN>
<SPAN style="color:#33cc00">t      </SPAN>  ->  <SPAN style="color:#33cc00">bright-green        </SPAN> <SPAN style="color:#aa6666">#turnlike</SPAN>
<SPAN style="color:#33cc00">u      </SPAN>  ->  <SPAN style="color:#33cc00">bright-green        </SPAN> <SPAN style="color:#aa6666">#tiny</SPAN>
</PRE><p>The default consensus colormap for nucleotide sequences can be selected using
the <code class="docutils literal notranslate"><span class="pre">-moltype</span> <span class="pre">na</span></code> (or <code class="docutils literal notranslate"><span class="pre">dna</span></code> or <code class="docutils literal notranslate"><span class="pre">rna</span></code>) option:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">moltype</span> <span class="n">na</span>
</pre></div>
</div>
<p>or by specifying:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">...</span> <span class="o">-</span><span class="n">con_colormap</span> <span class="n">DC1</span>
</pre></div>
</div>
<p>and is defined as:</p>
<PRE>
<SPAN style="color:#000000">[DC1]</SPAN>
<SPAN style="color:#aa6666">#DNA consensus: highlight ring type
#symbols =>  color                #comment</SPAN>
<SPAN style="color:#666666">.      </SPAN>  ->  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#unconserved</SPAN>
<SPAN style="color:#6600cc">r      </SPAN>  ->  <SPAN style="color:#6600cc">purple              </SPAN> <SPAN style="color:#aa6666">#purine</SPAN>
<SPAN style="color:#ff3333">y      </SPAN>  ->  <SPAN style="color:#ff3333">orange              </SPAN> <SPAN style="color:#aa6666">#pyrimidine</SPAN>
</PRE></section>
<section id="creating-new-colourmaps">
<h3>Creating new colourmaps<a class="headerlink" href="#creating-new-colourmaps" title="Permalink to this headline">¶</a></h3>
<p>The built-in colour palette and colourmaps built from it can be listed from
the command line with <code class="docutils literal notranslate"><span class="pre">-listcolors</span></code>, and new colour schemes can be loaded
from a file using the <code class="docutils literal notranslate"><span class="pre">-colorfile</span></code> option.</p>
<p>Predefined colours are defined as in the following short segment of the built
in colour palette obtained with <code class="docutils literal notranslate"><span class="pre">-listcolors</span> <span class="pre">-html</span> <span class="pre">head</span></code> to wrap the output
in HTML and display the actual colours:</p>
<PRE>
<SPAN style="color:#aa6666">#color                     : #RGB</SPAN>
color <SPAN style="color:#000000">black               </SPAN> : #000000
color <SPAN style="color:#ffffff">white               </SPAN> : #ffffff
color <SPAN style="color:#ff0000">red                 </SPAN> : #ff0000
color <SPAN style="color:#00ff00">green               </SPAN> : #00ff00
color <SPAN style="color:#0000ff">blue                </SPAN> : #0000ff
color <SPAN style="color:#00ffff">cyan                </SPAN> : #00ffff
color <SPAN style="color:#ff00ff">magenta             </SPAN> : #ff00ff
color <SPAN style="color:#ffff00">yellow              </SPAN> : #ffff00
...
</PRE><p>Here’s an example of a short protein colouring scheme using the built-in
colourmap, which is used to explain the syntax:</p>
<PRE>
<SPAN style="color:#000000">[CYS]</SPAN>
<SPAN style="color:#aa6666">#Protein: highlight cysteines
#symbols =>  color                #comment</SPAN>
<SPAN style="color:#666666">.      </SPAN>  ->  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#wildcard/mismatch</SPAN>
<SPAN style="color:#ffff00">Cc     </SPAN>  =>  <SPAN style="color:#ffff00">yellow              </SPAN> <SPAN style="color:#aa6666">#cysteine</SPAN>
<SPAN style="color:#666666">Xx     </SPAN>  ->  <SPAN style="color:#666666">dark-gray           </SPAN> <SPAN style="color:#aa6666">#unknown</SPAN>
<SPAN style="color:#999999">?      </SPAN>  ->  <SPAN style="color:#999999">light-gray          </SPAN> <SPAN style="color:#aa6666">#unknown</SPAN>
</PRE><p>When writing a new protein/nucleotide colouring scheme, scheme names
introduced in square brackets (<code class="docutils literal notranslate"><span class="pre">[CYS]</span></code>, above) are case-insensitive.</p>
<p>Any line or part of a line beginning with a <code class="docutils literal notranslate"><span class="pre">#</span></code> character is a comment.</p>
<p>Colourings are defined one per line by symbol(s) at the left, an arrow, then
the colour name or RGB code, followed by an optional comment. The symbols at
the left are case-sensitive, and can be given as single characters <code class="docutils literal notranslate"><span class="pre">X</span></code> or as
a character pair like <code class="docutils literal notranslate"><span class="pre">Xx</span></code>, where we want both upper- and lowercase to have
the same colour. The special wildcard <code class="docutils literal notranslate"><span class="pre">.</span></code> symbol sets the base colour to be
used for all symbols in the alignment. Other lines define specific colourings
for sequence symbols of interest. The ordering of lines is not important.</p>
<p>In the example, <code class="docutils literal notranslate"><span class="pre">C</span></code> or <code class="docutils literal notranslate"><span class="pre">c</span></code> will be painted yellow; an explicit <code class="docutils literal notranslate"><span class="pre">X</span></code> or
<code class="docutils literal notranslate"><span class="pre">x</span></code> residue will be dark grey; an explicit unknown residue <code class="docutils literal notranslate"><span class="pre">?</span></code> will be
light gray; any other residue will match the <code class="docutils literal notranslate"><span class="pre">.</span></code> wildcard and be painted
dark grey.</p>
<p>A symbol to name mapping can use a predefined colour name (as above) or an
explicit hexadecimal RGB code like those in the colour palette.</p>
<p>The arrow separating the symbol(s) from the colour code can be double <code class="docutils literal notranslate"><span class="pre">=&gt;</span></code>
or single <code class="docutils literal notranslate"><span class="pre">-&gt;</span></code> arrows selecting background or foreground colouring:</p>
<ul class="simple">
<li><p>If style sheets are not being used, the choice of arrow is unimportant: the
supplied colour is used for the foreground, i.e., the output symbol.</p></li>
<li><p>If style sheets are in use with <code class="docutils literal notranslate"><span class="pre">-css</span> <span class="pre">on</span></code>, then <code class="docutils literal notranslate"><span class="pre">=&gt;</span></code> means that the
colour should be applied to the background of the symbol, while <code class="docutils literal notranslate"><span class="pre">-&gt;</span></code> means
it should be aplied to the foreground, i.e., the symbol itself.</p></li>
</ul>
<p>So, in the <code class="docutils literal notranslate"><span class="pre">CYS</span></code> example, the symbols <code class="docutils literal notranslate"><span class="pre">*</span></code>, <code class="docutils literal notranslate"><span class="pre">?</span></code>, <code class="docutils literal notranslate"><span class="pre">X</span></code>, <code class="docutils literal notranslate"><span class="pre">x</span></code> will be
coloured in the foreground (i.e., the symbols themselves), and <code class="docutils literal notranslate"><span class="pre">C</span></code> or <code class="docutils literal notranslate"><span class="pre">c</span></code>
will be displayed as coloured symbols unless <code class="docutils literal notranslate"><span class="pre">-css</span> <span class="pre">on</span></code> is set, in which case
they will appear as coloured bocks.</p>
</section>
</section>
<section id="coverage-and-identity">
<span id="ref-cov-pid"></span><h2>Coverage and identity<a class="headerlink" href="#coverage-and-identity" title="Permalink to this headline">¶</a></h2>
<p>MView calculates percent coverage and percent identity of every sequence with
respect to a reference sequence, by default the query sequence of a search, or
the first row of a sequence alignment.</p>
<p>You can change the reference sequence against which identities are calculated
using the <code class="docutils literal notranslate"><span class="pre">-reference</span></code> option, which requires either a row number or a
sequence identifier (see <a class="reference internal" href="#ref-reference-row"><span class="std std-ref">Changing the reference sequence</span></a>).</p>
<p>In addition, you can resort the alignment using the <code class="docutils literal notranslate"><span class="pre">-sort</span></code> option (see
<a class="reference internal" href="#ref-sorting"><span class="std std-ref">Sorting by percent coverage or percent identity</span></a>).</p>
<p>The coverage and identity calculations are defined below.</p>
<section id="percent-coverage">
<h3>Percent coverage<a class="headerlink" href="#percent-coverage" title="Permalink to this headline">¶</a></h3>
<p>Percent coverages reported in each alignment row are calculated with respect to
the reference sequence (by default, the query or first row):</p>
<div class="math notranslate nohighlight">
\[\frac{\mathrm{number~of~residues~in~row~aligned~with~reference~row}}
     {\mathrm{length~of~ungapped~reference~row}}
\times 100\]</div>
</section>
<section id="percent-identity">
<h3>Percent identity<a class="headerlink" href="#percent-identity" title="Permalink to this headline">¶</a></h3>
<p>Percent identities reported in each alignment row are calculated with respect
to the aligned portion of the reference sequence (usually the query or first
row):</p>
<div class="math notranslate nohighlight">
\[\frac{\mathrm{number~of~identical~residues}}
     {\mathrm{length~of~ungapped~reference~row~over~aligned~region}}
\times 100\]</div>
<p>Two other calculation possibilities are available: <code class="docutils literal notranslate"><span class="pre">-pcid</span> <span class="pre">reference</span></code>
normalises by the ungapped length of the query or reference sequence, and
<code class="docutils literal notranslate"><span class="pre">-pcid</span> <span class="pre">hit</span></code> normalises by the ungapped length of the hit sequence.</p>
<p>Note: in the case of BLAST MView output, minor deviations from the percentages
reported by BLAST are due to (1) different rounding, and (2) the way MView
assembles a single pseudo-sequence (see <a class="reference internal" href="faq.html#ref-funny-sequences"><span class="std std-ref">Why are some sequence symbols lowercased?</span></a>) for a hit
composed of multiple HSPs, giving an averaged percent identity. This default
behaviour above is also obtained using the option <code class="docutils literal notranslate"><span class="pre">-pcid</span> <span class="pre">aligned</span></code>.</p>
</section>
</section>
<section id="data-formats">
<h2>Data formats<a class="headerlink" href="#data-formats" title="Permalink to this headline">¶</a></h2>
<section id="input-formats">
<h3>Input formats<a class="headerlink" href="#input-formats" title="Permalink to this headline">¶</a></h3>
<p>MView supports a variety of input formats covering common sequence database
seach and multiple alignment formats. Alternatively, if you can convert some
strange alignment to one of the simpler input formats (FASTA, PIR, MSF, plain)
you can then read it into MView. See <a class="reference external" href="formats_in.html">input_formats</a>.</p>
</section>
<section id="output-formats">
<h3>Output formats<a class="headerlink" href="#output-formats" title="Permalink to this headline">¶</a></h3>
<p>The default output is plain text showing the alignment together with some
header information. HTML markup will be added if any of the HTML-specific or
colouring options are set.</p>
<p>However, a number of alternative output formats allow format conversions
(e.g., convert a BLAST search to FASTA sequence format) for subsequent
processing. See <a class="reference external" href="formats_out.html">output_formats</a>.</p>
</section>
</section>
<section id="linking-identifiers-to-external-resources">
<h2>Linking identifiers to external resources<a class="headerlink" href="#linking-identifiers-to-external-resources" title="Permalink to this headline">¶</a></h2>
<p>Using the <code class="docutils literal notranslate"><span class="pre">-srs</span> <span class="pre">on</span></code> option with HTML output, it is possible to convert
sequence identifiers into links to a sequence database:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mview</span> <span class="o">-</span><span class="ow">in</span> <span class="n">fasta</span> <span class="o">-</span><span class="n">html</span> <span class="n">head</span> <span class="o">-</span><span class="n">css</span> <span class="n">on</span> <span class="o">-</span><span class="n">coloring</span> <span class="nb">id</span> <span class="o">-</span><span class="n">colormap</span> <span class="n">clustal</span> <span class="o">-</span><span class="n">srs</span> <span class="n">on</span> <span class="n">data</span><span class="o">.</span><span class="n">dat</span>
</pre></div>
</div>
<PRE>
Colored by: identity
                             cov    pid  1 [        .         .         .         .         :         .         .         ] 80
1 <A HREF="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?val=P00533">sp|P00533.2|EGFR_HUMAN</A>  100.0% 100.0%    <SPAN CLASS=S38>F</SPAN><SPAN CLASS=S37>KK</SPAN><SPAN CLASS=S38>I</SPAN><SPAN CLASS=S37>K</SPAN><SPAN CLASS=S38>VL</SPAN><SPAN CLASS=S44>G</SPAN><SPAN CLASS=S39>S</SPAN><SPAN CLASS=S44>G</SPAN><SPAN CLASS=S38>AF</SPAN><SPAN CLASS=S44>G</SPAN><SPAN CLASS=S39>T</SPAN><SPAN CLASS=S38>V</SPAN><SPAN CLASS=S40>Y</SPAN><SPAN CLASS=S37>K</SPAN><SPAN CLASS=S44>G</SPAN><SPAN CLASS=S38>LWI</SPAN><SPAN CLASS=S43>P</SPAN><SPAN CLASS=S42>E</SPAN><SPAN CLASS=S44>G</SPAN><SPAN CLASS=S42>E</SPAN><SPAN CLASS=S37>K</SPAN><SPAN style="color:#666666">---------</SPAN><SPAN CLASS=S38>V</SPAN><SPAN CLASS=S37>K</SPAN><SPAN CLASS=S38>I</SPAN><SPAN CLASS=S43>P</SPAN><SPAN CLASS=S38>VAI</SPAN><SPAN CLASS=S37>K</SPAN><SPAN CLASS=S42>E</SPAN><SPAN CLASS=S38>L</SPAN><SPAN CLASS=S37>R</SPAN><SPAN CLASS=S42>E</SPAN><SPAN CLASS=S38>A</SPAN><SPAN CLASS=S39>TS</SPAN><SPAN CLASS=S43>P</SPAN><SPAN CLASS=S37>K</SPAN><SPAN style="color:#666666">-</SPAN><SPAN CLASS=S38>A</SPAN><SPAN CLASS=S39>N</SPAN><SPAN CLASS=S37>K</SPAN><SPAN CLASS=S42>E</SPAN><SPAN CLASS=S38>IL</SPAN><SPAN CLASS=S42>DE</SPAN><SPAN CLASS=S38>A</SPAN><SPAN CLASS=S40>Y</SPAN><SPAN CLASS=S38>VMA</SPAN><SPAN CLASS=S39>S</SPAN><SPAN CLASS=S38>V</SPAN><SPAN CLASS=S42>D</SPAN><SPAN CLASS=S39>N</SPAN><SPAN CLASS=S43>P</SPAN><SPAN CLASS=S40>H</SPAN><SPAN CLASS=S38>V</SPAN><SPAN CLASS=S41>C</SPAN><SPAN CLASS=S37>R</SPAN><SPAN CLASS=S38>LL</SPAN><SPAN CLASS=S44>G</SPAN><SPAN CLASS=S38>I</SPAN><SPAN CLASS=S41>C</SPAN>
2 <A HREF="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?val=Q9I7F7">sp|Q9I7F7.3|PR2_DROME</A>    97.1%  35.7%    <SPAN style="color:#666666">ISVN</SPAN><SPAN CLASS=S37>K</SPAN><SPAN style="color:#666666">Q</SPAN><SPAN CLASS=S38>L</SPAN><SPAN CLASS=S44>G</SPAN><SPAN style="color:#666666">T</SPAN><SPAN CLASS=S44>G</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S38>F</SPAN><SPAN CLASS=S44>G</SPAN><SPAN style="color:#666666">I</SPAN><SPAN CLASS=S38>V</SPAN><SPAN style="color:#666666">QQ</SPAN><SPAN CLASS=S44>G</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S38>W</SPAN><SPAN style="color:#666666">SNGN</SPAN><SPAN CLASS=S42>E</SPAN><SPAN style="color:#666666">-----------R</SPAN><SPAN CLASS=S38>I</SPAN><SPAN style="color:#666666">Q</SPAN><SPAN CLASS=S38>VAI</SPAN><SPAN CLASS=S37>K</SPAN><SPAN style="color:#666666">C</SPAN><SPAN CLASS=S38>L</SPAN><SPAN style="color:#666666">CRERMQS-NPM</SPAN><SPAN CLASS=S42>E</SPAN><SPAN style="color:#666666">F</SPAN><SPAN CLASS=S38>L</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S42>E</SPAN><SPAN CLASS=S38>A</SPAN><SPAN style="color:#666666">AI</SPAN><SPAN CLASS=S38>M</SPAN><SPAN style="color:#666666">H</SPAN><SPAN CLASS=S39>S</SPAN><SPAN style="color:#666666">IEHENIV</SPAN><SPAN CLASS=S37>R</SPAN><SPAN CLASS=S38>L</SPAN><SPAN style="color:#666666">Y</SPAN><SPAN CLASS=S44>G</SPAN><SPAN style="color:#666666">VV</SPAN>
3 <A HREF="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?val=Q08881">sp|Q08881.1|ITK_HUMAN</A>    90.0%  32.9%    <SPAN style="color:#666666">LTFVQEI</SPAN><SPAN CLASS=S44>G</SPAN><SPAN CLASS=S39>S</SPAN><SPAN CLASS=S44>G</SPAN><SPAN style="color:#666666">Q</SPAN><SPAN CLASS=S38>F</SPAN><SPAN CLASS=S44>G</SPAN><SPAN style="color:#666666">L</SPAN><SPAN CLASS=S38>V</SPAN><SPAN style="color:#666666">HL</SPAN><SPAN CLASS=S44>G</SPAN><SPAN style="color:#666666">Y</SPAN><SPAN CLASS=S38>W</SPAN><SPAN style="color:#666666">LN--------------</SPAN><SPAN CLASS=S37>K</SPAN><SPAN style="color:#666666">DK</SPAN><SPAN CLASS=S38>VAI</SPAN><SPAN CLASS=S37>K</SPAN><SPAN style="color:#666666">TI</SPAN><SPAN CLASS=S37>R</SPAN><SPAN CLASS=S42>E</SPAN><SPAN style="color:#666666">GAMS---EEDFIE</SPAN><SPAN CLASS=S42>E</SPAN><SPAN CLASS=S38>A</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S38>VM</SPAN><SPAN style="color:#666666">MKLSH</SPAN><SPAN CLASS=S43>P</SPAN><SPAN style="color:#666666">KLVQ</SPAN><SPAN CLASS=S38>L</SPAN><SPAN style="color:#666666">Y</SPAN><SPAN CLASS=S44>G</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S41>C</SPAN>
4 <A HREF="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?val=Q13308">sp|Q13308.2|PTK7_HUMAN</A>   97.1%  21.2%    <SPAN style="color:#666666">IREV</SPAN><SPAN CLASS=S37>K</SPAN><SPAN style="color:#666666">QI</SPAN><SPAN CLASS=S44>G</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S44>G</SPAN><SPAN style="color:#666666">Q</SPAN><SPAN CLASS=S38>F</SPAN><SPAN CLASS=S44>G</SPAN><SPAN style="color:#666666">A</SPAN><SPAN CLASS=S38>V</SPAN><SPAN style="color:#666666">VLAEMTGLS-XLPKGSMNADGVAL</SPAN><SPAN CLASS=S38>VA</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S37>K</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S38>L</SPAN><SPAN style="color:#666666">KPDV</SPAN><SPAN CLASS=S39>S</SPAN><SPAN style="color:#666666">D-EVLQSFDK</SPAN><SPAN CLASS=S42>E</SPAN><SPAN style="color:#666666">IKF</SPAN><SPAN CLASS=S38>M</SPAN><SPAN style="color:#666666">SQLQHDSIVQ</SPAN><SPAN CLASS=S38>LL</SPAN><SPAN style="color:#666666">A</SPAN><SPAN CLASS=S38>I</SPAN><SPAN CLASS=S41>C</SPAN>
5 <A HREF="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?val=P34265">sp|P34265.4|KIN31_CAEEL</A> 100.0%  31.5%    <SPAN style="color:#666666">VELT</SPAN><SPAN CLASS=S37>K</SPAN><SPAN style="color:#666666">K</SPAN><SPAN CLASS=S38>L</SPAN><SPAN CLASS=S44>G</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S44>G</SPAN><SPAN CLASS=S38>AF</SPAN><SPAN CLASS=S44>G</SPAN><SPAN style="color:#666666">E</SPAN><SPAN CLASS=S38>V</SPAN><SPAN style="color:#666666">W</SPAN><SPAN CLASS=S37>K</SPAN><SPAN CLASS=S44>G</SPAN><SPAN style="color:#666666">KLLKILDA-------NHQPVL</SPAN><SPAN CLASS=S38>VA</SPAN><SPAN style="color:#666666">V</SPAN><SPAN CLASS=S37>K</SPAN><SPAN style="color:#666666">TAKLESMT</SPAN><SPAN CLASS=S37>K</SPAN><SPAN style="color:#666666">EQI</SPAN><SPAN CLASS=S37>K</SPAN><SPAN CLASS=S42>E</SPAN><SPAN CLASS=S38>I</SPAN><SPAN style="color:#666666">MR</SPAN><SPAN CLASS=S42>E</SPAN><SPAN CLASS=S38>A</SPAN><SPAN style="color:#666666">RL</SPAN><SPAN CLASS=S38>M</SPAN><SPAN style="color:#666666">RNL</SPAN><SPAN CLASS=S42>D</SPAN><SPAN style="color:#666666">HIN</SPAN><SPAN CLASS=S38>V</SPAN><SPAN style="color:#666666">VKFF</SPAN><SPAN CLASS=S44>G</SPAN><SPAN style="color:#666666">VA</SPAN>
</PRE><p>The identfiers need to conform to patterns such as:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">database</span><span class="o">|</span><span class="n">accession</span><span class="o">|</span><span class="n">identifier</span>
<span class="n">database</span><span class="p">:</span><span class="n">identifier</span>
</pre></div>
</div>
<p>like those produced by the NCBI, EBI or other blast services.</p>
<p>Links will be constructed if the patterns are listed in the <code class="docutils literal notranslate"><span class="pre">SRS.pm</span></code>
library, which is part of this software. You can modify and extend this file
to include more patterns if you know some Perl and the format of the URLs
needed to access the sequence databases of interest.</p>
<p>Of course, this linking mechanism works for any recognised input data format,
not just blast results.</p>
</section>
<section id="memory-usage-and-speed">
<h2>Memory usage and speed<a class="headerlink" href="#memory-usage-and-speed" title="Permalink to this headline">¶</a></h2>
<p>Use of memory by MView can be very great, particularly if you try to process
complete sets of PSI-BLAST cycles each containing 1000s of hits all at
once. Use of most filtering options should reduce memory requirements by
cutting down the number of internal data structures created.</p>
<p>In particular, if you only want to see the scoring information and don’t care
about the sequence alignments, you can switch these off with the <code class="docutils literal notranslate"><span class="pre">-sequences</span>
<span class="pre">off</span></code> option, which will also speed up the program.</p>
<p>Likewise, processing each alignment (for example, plus strands and then minus
strands) separately will save memory, or you can use the option <code class="docutils literal notranslate"><span class="pre">-register</span>
<span class="pre">off</span></code> to cause each alignment to be output when ready (by default all
alignments are saved until the end so they can be printed with fields in
vertical register).</p>
</section>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="../index.html">Table of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Manual</a><ul>
<li><a class="reference internal" href="#basic-usage">Basic usage</a><ul>
<li><a class="reference internal" href="#getting-started">Getting started</a></li>
<li><a class="reference internal" href="#the-ruler-line">The ruler line</a></li>
<li><a class="reference internal" href="#command-line-options">Command line options</a></li>
</ul>
</li>
<li><a class="reference internal" href="#layout-and-filtering">Layout and filtering</a><ul>
<li><a class="reference internal" href="#pagination">Pagination</a></li>
<li><a class="reference internal" href="#column-ranges">Column ranges</a></li>
<li><a class="reference internal" href="#changing-the-reference-sequence">Changing the reference sequence</a></li>
<li><a class="reference internal" href="#sorting-by-percent-coverage-or-percent-identity">Sorting by percent coverage or percent identity</a></li>
<li><a class="reference internal" href="#filtering-rows">Filtering rows</a></li>
<li><a class="reference internal" href="#labels-annotations-and-sequences">Labels, annotations and sequences</a></li>
</ul>
</li>
<li><a class="reference internal" href="#adding-html">Adding HTML</a><ul>
<li><a class="reference internal" href="#basic-html">Basic HTML</a></li>
<li><a class="reference internal" href="#controlling-the-amount-of-html">Controlling the amount of HTML</a></li>
<li><a class="reference internal" href="#using-an-external-css-style-sheet">Using an external CSS style sheet</a></li>
</ul>
</li>
<li><a class="reference internal" href="#consensus-sequences">Consensus sequences</a><ul>
<li><a class="reference internal" href="#clustal-conservation-line">Clustal conservation line</a></li>
<li><a class="reference internal" href="#consensus-lines">Consensus lines</a></li>
<li><a class="reference internal" href="#changing-consensus-thresholds">Changing consensus thresholds</a></li>
<li><a class="reference internal" href="#consensus-pattern-definitions">Consensus pattern definitions</a></li>
<li><a class="reference internal" href="#changing-consensus-patterns">Changing consensus patterns</a></li>
<li><a class="reference internal" href="#showing-conserved-residues-or-conserved-classes">Showing conserved residues or conserved classes</a></li>
</ul>
</li>
<li><a class="reference internal" href="#colouring-modes">Colouring modes</a><ul>
<li><a class="reference internal" href="#alignment-colouring">Alignment colouring</a></li>
<li><a class="reference internal" href="#conserved-residue-colouring">Conserved residue colouring</a></li>
<li><a class="reference internal" href="#finding-and-colouring-motifs">Finding and colouring motifs</a></li>
<li><a class="reference internal" href="#combining-alignment-and-motif-colouring">Combining alignment and motif colouring</a></li>
<li><a class="reference internal" href="#comment-row-colouring">Comment row colouring</a></li>
<li><a class="reference internal" href="#consensus-pattern-colouring">Consensus pattern colouring</a></li>
</ul>
</li>
<li><a class="reference internal" href="#colours">Colours</a><ul>
<li><a class="reference internal" href="#colourmaps">Colourmaps</a></li>
<li><a class="reference internal" href="#creating-new-colourmaps">Creating new colourmaps</a></li>
</ul>
</li>
<li><a class="reference internal" href="#coverage-and-identity">Coverage and identity</a><ul>
<li><a class="reference internal" href="#percent-coverage">Percent coverage</a></li>
<li><a class="reference internal" href="#percent-identity">Percent identity</a></li>
</ul>
</li>
<li><a class="reference internal" href="#data-formats">Data formats</a><ul>
<li><a class="reference internal" href="#input-formats">Input formats</a></li>
<li><a class="reference internal" href="#output-formats">Output formats</a></li>
</ul>
</li>
<li><a class="reference internal" href="#linking-identifiers-to-external-resources">Linking identifiers to external resources</a></li>
<li><a class="reference internal" href="#memory-usage-and-speed">Memory usage and speed</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="../contents.html"
                        title="previous chapter">Contents</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="formats_in.html"
                        title="next chapter">Input formats</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="../_sources/manual/manual.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="formats_in.html" title="Input formats"
             >next</a> |</li>
        <li class="right" >
          <a href="../contents.html" title="Contents"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">MView</a> &#187;</li>
          <li class="nav-item nav-item-1"><a href="../contents.html" >Contents</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Manual</a></li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2015-2020, Nigel P. Brown.
      Last updated on 08 Sep 2024.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 4.3.2.
    </div>
  </body>
</html>